{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4a8d1386",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mDEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621\u001b[0m\u001b[33m\n",
      "\u001b[0mLooking in indexes: https://pypi.org/simple, https://alexander.ukhatov:****@pypi.pkg.jetbrains.space/glassdomeinc/p/ai/forecast-anomaly-data-science/simple\n",
      "Requirement already satisfied: openpyxl in /usr/local/lib/python3.9/site-packages (3.1.2)\n",
      "Requirement already satisfied: et-xmlfile in /usr/local/lib/python3.9/site-packages (from openpyxl) (1.1.0)\n",
      "\u001b[33mDEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621\u001b[0m\u001b[33m\n",
      "\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip available: \u001b[0m\u001b[31;49m22.3.1\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m23.2.1\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpython3.9 -m pip install --upgrade pip\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sys \n",
    "!{sys.executable} -m pip install openpyxl\n",
    "import openpyxl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "863fbb02",
   "metadata": {},
   "outputs": [],
   "source": [
    "cgga_data_mRNAseq = pd.read_csv(\"CGGA.mRNAseq_325.RSEM-genes.20200506.txt\", sep='\\t')\n",
    "cgga_data_mRNAseq.set_index(\"Gene_Name\", inplace=True)\n",
    "cgga_data_mRNAseq = cgga_data_mRNAseq.transpose()\n",
    "cgga_data_mRNAseq.index.name = \"Sample\"\n",
    "# cgga_data.set_index(\"Sample\", inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "cb4cc122",
   "metadata": {},
   "outputs": [],
   "source": [
    "# cgga_data_mRNAseq.to_csv('cgga_data_mRNAseq.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "4b36210e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene_Name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2M-AS1</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A2MP1</th>\n",
       "      <th>A3GALT2</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>A4GNT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>...</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDA</th>\n",
       "      <th>ZXDB</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11A</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_1001</th>\n",
       "      <td>12.64</td>\n",
       "      <td>2.12</td>\n",
       "      <td>452.92</td>\n",
       "      <td>3.30</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.67</td>\n",
       "      <td>0.04</td>\n",
       "      <td>27.79</td>\n",
       "      <td>...</td>\n",
       "      <td>3.10</td>\n",
       "      <td>2.26</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.33</td>\n",
       "      <td>8.44</td>\n",
       "      <td>0.02</td>\n",
       "      <td>3.81</td>\n",
       "      <td>158.36</td>\n",
       "      <td>6.56</td>\n",
       "      <td>6.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1006</th>\n",
       "      <td>7.03</td>\n",
       "      <td>1.13</td>\n",
       "      <td>106.54</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.00</td>\n",
       "      <td>3.16</td>\n",
       "      <td>1.07</td>\n",
       "      <td>24.15</td>\n",
       "      <td>...</td>\n",
       "      <td>9.47</td>\n",
       "      <td>20.55</td>\n",
       "      <td>1.41</td>\n",
       "      <td>2.99</td>\n",
       "      <td>15.59</td>\n",
       "      <td>0.49</td>\n",
       "      <td>7.84</td>\n",
       "      <td>761.56</td>\n",
       "      <td>5.86</td>\n",
       "      <td>20.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1007</th>\n",
       "      <td>30.09</td>\n",
       "      <td>6.64</td>\n",
       "      <td>206.70</td>\n",
       "      <td>0.63</td>\n",
       "      <td>4.96</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0.17</td>\n",
       "      <td>6.67</td>\n",
       "      <td>0.00</td>\n",
       "      <td>26.03</td>\n",
       "      <td>...</td>\n",
       "      <td>3.03</td>\n",
       "      <td>16.88</td>\n",
       "      <td>2.04</td>\n",
       "      <td>2.87</td>\n",
       "      <td>9.53</td>\n",
       "      <td>0.21</td>\n",
       "      <td>8.09</td>\n",
       "      <td>92.85</td>\n",
       "      <td>9.78</td>\n",
       "      <td>11.42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1011</th>\n",
       "      <td>6.64</td>\n",
       "      <td>4.32</td>\n",
       "      <td>707.17</td>\n",
       "      <td>1.61</td>\n",
       "      <td>1.59</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2.07</td>\n",
       "      <td>0.11</td>\n",
       "      <td>35.62</td>\n",
       "      <td>...</td>\n",
       "      <td>5.27</td>\n",
       "      <td>5.14</td>\n",
       "      <td>1.08</td>\n",
       "      <td>1.91</td>\n",
       "      <td>9.28</td>\n",
       "      <td>0.04</td>\n",
       "      <td>5.68</td>\n",
       "      <td>158.75</td>\n",
       "      <td>11.48</td>\n",
       "      <td>15.82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1015</th>\n",
       "      <td>1.83</td>\n",
       "      <td>1.39</td>\n",
       "      <td>824.32</td>\n",
       "      <td>1.34</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.30</td>\n",
       "      <td>0.16</td>\n",
       "      <td>9.41</td>\n",
       "      <td>0.13</td>\n",
       "      <td>35.83</td>\n",
       "      <td>...</td>\n",
       "      <td>3.71</td>\n",
       "      <td>9.43</td>\n",
       "      <td>1.50</td>\n",
       "      <td>2.38</td>\n",
       "      <td>7.46</td>\n",
       "      <td>0.02</td>\n",
       "      <td>7.04</td>\n",
       "      <td>380.73</td>\n",
       "      <td>11.71</td>\n",
       "      <td>13.35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_243</th>\n",
       "      <td>4.03</td>\n",
       "      <td>2.47</td>\n",
       "      <td>108.06</td>\n",
       "      <td>0.33</td>\n",
       "      <td>1.55</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.85</td>\n",
       "      <td>0.00</td>\n",
       "      <td>29.28</td>\n",
       "      <td>...</td>\n",
       "      <td>2.96</td>\n",
       "      <td>7.05</td>\n",
       "      <td>4.00</td>\n",
       "      <td>7.37</td>\n",
       "      <td>10.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>10.26</td>\n",
       "      <td>21.45</td>\n",
       "      <td>14.00</td>\n",
       "      <td>8.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_247</th>\n",
       "      <td>5.29</td>\n",
       "      <td>3.69</td>\n",
       "      <td>97.24</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.71</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.27</td>\n",
       "      <td>1.20</td>\n",
       "      <td>0.73</td>\n",
       "      <td>45.31</td>\n",
       "      <td>...</td>\n",
       "      <td>6.53</td>\n",
       "      <td>61.94</td>\n",
       "      <td>3.34</td>\n",
       "      <td>5.67</td>\n",
       "      <td>16.88</td>\n",
       "      <td>0.25</td>\n",
       "      <td>8.81</td>\n",
       "      <td>67.38</td>\n",
       "      <td>17.09</td>\n",
       "      <td>21.31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_738</th>\n",
       "      <td>4.64</td>\n",
       "      <td>0.85</td>\n",
       "      <td>160.70</td>\n",
       "      <td>0.38</td>\n",
       "      <td>3.33</td>\n",
       "      <td>0.11</td>\n",
       "      <td>0.54</td>\n",
       "      <td>2.69</td>\n",
       "      <td>1.37</td>\n",
       "      <td>38.59</td>\n",
       "      <td>...</td>\n",
       "      <td>2.85</td>\n",
       "      <td>6.89</td>\n",
       "      <td>2.60</td>\n",
       "      <td>3.89</td>\n",
       "      <td>12.87</td>\n",
       "      <td>0.53</td>\n",
       "      <td>9.91</td>\n",
       "      <td>72.42</td>\n",
       "      <td>15.13</td>\n",
       "      <td>12.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_759</th>\n",
       "      <td>7.07</td>\n",
       "      <td>1.99</td>\n",
       "      <td>263.66</td>\n",
       "      <td>0.48</td>\n",
       "      <td>0.60</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2.62</td>\n",
       "      <td>0.16</td>\n",
       "      <td>34.63</td>\n",
       "      <td>...</td>\n",
       "      <td>5.16</td>\n",
       "      <td>27.59</td>\n",
       "      <td>1.67</td>\n",
       "      <td>2.53</td>\n",
       "      <td>8.99</td>\n",
       "      <td>0.00</td>\n",
       "      <td>7.53</td>\n",
       "      <td>86.75</td>\n",
       "      <td>12.51</td>\n",
       "      <td>11.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_D30</th>\n",
       "      <td>31.41</td>\n",
       "      <td>2.65</td>\n",
       "      <td>192.54</td>\n",
       "      <td>2.51</td>\n",
       "      <td>3.86</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.19</td>\n",
       "      <td>3.36</td>\n",
       "      <td>0.00</td>\n",
       "      <td>24.39</td>\n",
       "      <td>...</td>\n",
       "      <td>1.89</td>\n",
       "      <td>6.30</td>\n",
       "      <td>1.29</td>\n",
       "      <td>2.81</td>\n",
       "      <td>7.16</td>\n",
       "      <td>0.02</td>\n",
       "      <td>7.49</td>\n",
       "      <td>85.42</td>\n",
       "      <td>4.20</td>\n",
       "      <td>8.31</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>325 rows × 24326 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Gene_Name   A1BG  A1BG-AS1     A2M  A2M-AS1  A2ML1  A2MP1  A3GALT2  A4GALT  \\\n",
       "Sample                                                                       \n",
       "CGGA_1001  12.64      2.12  452.92     3.30   0.04   0.23     0.00    3.67   \n",
       "CGGA_1006   7.03      1.13  106.54     0.13   0.33   0.00     1.00    3.16   \n",
       "CGGA_1007  30.09      6.64  206.70     0.63   4.96   0.15     0.17    6.67   \n",
       "CGGA_1011   6.64      4.32  707.17     1.61   1.59   0.14     0.00    2.07   \n",
       "CGGA_1015   1.83      1.39  824.32     1.34   0.00   1.30     0.16    9.41   \n",
       "...          ...       ...     ...      ...    ...    ...      ...     ...   \n",
       "CGGA_243    4.03      2.47  108.06     0.33   1.55   0.19     0.00    0.85   \n",
       "CGGA_247    5.29      3.69   97.24     0.03   0.71   0.14     0.27    1.20   \n",
       "CGGA_738    4.64      0.85  160.70     0.38   3.33   0.11     0.54    2.69   \n",
       "CGGA_759    7.07      1.99  263.66     0.48   0.60   0.02     0.00    2.62   \n",
       "CGGA_D30   31.41      2.65  192.54     2.51   3.86   0.09     0.19    3.36   \n",
       "\n",
       "Gene_Name  A4GNT   AAAS  ...  ZWILCH  ZWINT  ZXDA  ZXDB   ZXDC  ZYG11A  \\\n",
       "Sample                   ...                                             \n",
       "CGGA_1001   0.04  27.79  ...    3.10   2.26  0.99  1.33   8.44    0.02   \n",
       "CGGA_1006   1.07  24.15  ...    9.47  20.55  1.41  2.99  15.59    0.49   \n",
       "CGGA_1007   0.00  26.03  ...    3.03  16.88  2.04  2.87   9.53    0.21   \n",
       "CGGA_1011   0.11  35.62  ...    5.27   5.14  1.08  1.91   9.28    0.04   \n",
       "CGGA_1015   0.13  35.83  ...    3.71   9.43  1.50  2.38   7.46    0.02   \n",
       "...          ...    ...  ...     ...    ...   ...   ...    ...     ...   \n",
       "CGGA_243    0.00  29.28  ...    2.96   7.05  4.00  7.37  10.99    0.00   \n",
       "CGGA_247    0.73  45.31  ...    6.53  61.94  3.34  5.67  16.88    0.25   \n",
       "CGGA_738    1.37  38.59  ...    2.85   6.89  2.60  3.89  12.87    0.53   \n",
       "CGGA_759    0.16  34.63  ...    5.16  27.59  1.67  2.53   8.99    0.00   \n",
       "CGGA_D30    0.00  24.39  ...    1.89   6.30  1.29  2.81   7.16    0.02   \n",
       "\n",
       "Gene_Name  ZYG11B     ZYX  ZZEF1   ZZZ3  \n",
       "Sample                                   \n",
       "CGGA_1001    3.81  158.36   6.56   6.12  \n",
       "CGGA_1006    7.84  761.56   5.86  20.55  \n",
       "CGGA_1007    8.09   92.85   9.78  11.42  \n",
       "CGGA_1011    5.68  158.75  11.48  15.82  \n",
       "CGGA_1015    7.04  380.73  11.71  13.35  \n",
       "...           ...     ...    ...    ...  \n",
       "CGGA_243    10.26   21.45  14.00   8.75  \n",
       "CGGA_247     8.81   67.38  17.09  21.31  \n",
       "CGGA_738     9.91   72.42  15.13  12.23  \n",
       "CGGA_759     7.53   86.75  12.51  11.28  \n",
       "CGGA_D30     7.49   85.42   4.20   8.31  \n",
       "\n",
       "[325 rows x 24326 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cgga_data_mRNAseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "165bedbc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# [name for name in cgga_data.columns if \"GC0\" in name]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "66b1fa2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# \"FABP1\" in cgga_data.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "d3756ff9",
   "metadata": {},
   "outputs": [],
   "source": [
    "genes = list(set(list(pd.read_excel('gene_list.xlsx', sheet_name=0, header=None)[0]) +\\\n",
    "    list(pd.read_excel('gene_list.xlsx', sheet_name=1, header=None)[0])+\\\n",
    "    list(pd.read_excel('gene_list.xlsx', sheet_name=2, header=None)[0])))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "5c05bbf9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "87 genes:  ['EVPL', 'BAG3', 'SKIL', 'HTR1B', 'KCNJ12', 'APP', 'P2RY14', 'CFTR', 'FABP1', 'LRRK2', 'MYC', 'MDFI', 'TIE1', 'C11orf54', 'CD36', 'RAF1', 'RLBP1', 'PSEN2', 'GDF15', 'DDRGK1', 'MT-ND6', 'MPL', 'AKR1E2', 'MYO5C', 'PGAM2', 'PHLDB2', 'MT-ND3', 'MYO1D', 'PLP1', 'PENK', 'CKMT2', 'ESR1', 'ABCC3', 'SLC9A3R2', 'PPP5C', 'AQP1', 'CBX3', 'DLG4', 'GPR3', 'HLA-DQA1', 'DPY19L3', 'THBS4', 'PTGDR2', 'CCM2L', 'SLC12A7', 'PTH1R', 'PAX6', 'VAX1', 'BMP8A', 'FSHR', 'KRT31', 'ITPR3', 'SYNJ2', 'DDIT4L', 'MT-RNR2', 'CD3D', 'CHIT1', 'OXGR1', 'ETV5', 'HTR1F', 'ACTB', 'SPRY4', 'MET', 'SPSB1', 'GNG12', 'SHC3', 'ETV4', 'GEM', 'PIM1', 'ANXA2', 'CRHR2', 'KRT19', 'GRB14', 'C1QTNF7', 'THSD7A', 'CORO6', 'DMRT3', 'HNF4A', 'BCL2L11', 'IGF2BP3', 'ALDH8A1', 'ALDH1A3', 'KRT5', 'PREX2', 'TP53TG5', 'TUBB4A', 'NPBWR2']\n"
     ]
    }
   ],
   "source": [
    "print(len(genes), \"genes: \", genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "6bc034f8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Not in cgga_data_mRNAseq: ['FABP1', 'AQP1', 'FSHR', 'NPBWR2']\n"
     ]
    }
   ],
   "source": [
    "genes_in_cgga_data_mRNAseq = [g for g in genes if g in cgga_data_mRNAseq.columns]\n",
    "genes_not_in_cgga_data_mRNAseq = [g for g in genes if g not in cgga_data_mRNAseq.columns]\n",
    "print(f\"Not in cgga_data_mRNAseq: {genes_not_in_cgga_data_mRNAseq}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "d0faaf80",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene_Name</th>\n",
       "      <th>EVPL</th>\n",
       "      <th>BAG3</th>\n",
       "      <th>SKIL</th>\n",
       "      <th>HTR1B</th>\n",
       "      <th>KCNJ12</th>\n",
       "      <th>APP</th>\n",
       "      <th>P2RY14</th>\n",
       "      <th>CFTR</th>\n",
       "      <th>LRRK2</th>\n",
       "      <th>MYC</th>\n",
       "      <th>...</th>\n",
       "      <th>DMRT3</th>\n",
       "      <th>HNF4A</th>\n",
       "      <th>BCL2L11</th>\n",
       "      <th>IGF2BP3</th>\n",
       "      <th>ALDH8A1</th>\n",
       "      <th>ALDH1A3</th>\n",
       "      <th>KRT5</th>\n",
       "      <th>PREX2</th>\n",
       "      <th>TP53TG5</th>\n",
       "      <th>TUBB4A</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_1001</th>\n",
       "      <td>0.00</td>\n",
       "      <td>34.89</td>\n",
       "      <td>9.44</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.02</td>\n",
       "      <td>517.02</td>\n",
       "      <td>0.61</td>\n",
       "      <td>0.03</td>\n",
       "      <td>4.09</td>\n",
       "      <td>34.75</td>\n",
       "      <td>...</td>\n",
       "      <td>0.54</td>\n",
       "      <td>0.03</td>\n",
       "      <td>8.18</td>\n",
       "      <td>1.37</td>\n",
       "      <td>1.10</td>\n",
       "      <td>7.93</td>\n",
       "      <td>0.03</td>\n",
       "      <td>2.71</td>\n",
       "      <td>0.35</td>\n",
       "      <td>34.38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1006</th>\n",
       "      <td>2.65</td>\n",
       "      <td>26.14</td>\n",
       "      <td>17.93</td>\n",
       "      <td>1.10</td>\n",
       "      <td>1.00</td>\n",
       "      <td>905.51</td>\n",
       "      <td>1.55</td>\n",
       "      <td>1.17</td>\n",
       "      <td>2.88</td>\n",
       "      <td>42.67</td>\n",
       "      <td>...</td>\n",
       "      <td>2.10</td>\n",
       "      <td>0.35</td>\n",
       "      <td>4.09</td>\n",
       "      <td>12.77</td>\n",
       "      <td>2.97</td>\n",
       "      <td>1.60</td>\n",
       "      <td>2.44</td>\n",
       "      <td>7.85</td>\n",
       "      <td>1.01</td>\n",
       "      <td>4.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1007</th>\n",
       "      <td>0.22</td>\n",
       "      <td>25.38</td>\n",
       "      <td>7.79</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.18</td>\n",
       "      <td>546.25</td>\n",
       "      <td>3.12</td>\n",
       "      <td>0.02</td>\n",
       "      <td>1.15</td>\n",
       "      <td>10.16</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>10.54</td>\n",
       "      <td>4.69</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.23</td>\n",
       "      <td>11.78</td>\n",
       "      <td>0.19</td>\n",
       "      <td>39.15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1011</th>\n",
       "      <td>0.21</td>\n",
       "      <td>42.45</td>\n",
       "      <td>17.02</td>\n",
       "      <td>0.21</td>\n",
       "      <td>0.06</td>\n",
       "      <td>310.67</td>\n",
       "      <td>1.67</td>\n",
       "      <td>0.08</td>\n",
       "      <td>10.27</td>\n",
       "      <td>28.24</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>5.65</td>\n",
       "      <td>2.03</td>\n",
       "      <td>0.86</td>\n",
       "      <td>1.89</td>\n",
       "      <td>0.00</td>\n",
       "      <td>6.09</td>\n",
       "      <td>0.11</td>\n",
       "      <td>14.42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1015</th>\n",
       "      <td>0.02</td>\n",
       "      <td>23.33</td>\n",
       "      <td>18.76</td>\n",
       "      <td>0.29</td>\n",
       "      <td>0.23</td>\n",
       "      <td>228.95</td>\n",
       "      <td>0.32</td>\n",
       "      <td>0.01</td>\n",
       "      <td>2.46</td>\n",
       "      <td>15.46</td>\n",
       "      <td>...</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.58</td>\n",
       "      <td>3.63</td>\n",
       "      <td>0.50</td>\n",
       "      <td>36.68</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_243</th>\n",
       "      <td>0.00</td>\n",
       "      <td>10.89</td>\n",
       "      <td>14.30</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.03</td>\n",
       "      <td>527.60</td>\n",
       "      <td>0.95</td>\n",
       "      <td>1.76</td>\n",
       "      <td>2.25</td>\n",
       "      <td>36.63</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.79</td>\n",
       "      <td>0.06</td>\n",
       "      <td>1.55</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>8.55</td>\n",
       "      <td>0.00</td>\n",
       "      <td>45.83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_247</th>\n",
       "      <td>1.94</td>\n",
       "      <td>20.06</td>\n",
       "      <td>17.87</td>\n",
       "      <td>0.81</td>\n",
       "      <td>1.15</td>\n",
       "      <td>593.17</td>\n",
       "      <td>0.92</td>\n",
       "      <td>0.45</td>\n",
       "      <td>2.39</td>\n",
       "      <td>31.10</td>\n",
       "      <td>...</td>\n",
       "      <td>0.84</td>\n",
       "      <td>0.14</td>\n",
       "      <td>12.90</td>\n",
       "      <td>0.02</td>\n",
       "      <td>2.14</td>\n",
       "      <td>0.31</td>\n",
       "      <td>25.00</td>\n",
       "      <td>2.10</td>\n",
       "      <td>0.11</td>\n",
       "      <td>52.49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_738</th>\n",
       "      <td>2.62</td>\n",
       "      <td>29.40</td>\n",
       "      <td>12.14</td>\n",
       "      <td>1.27</td>\n",
       "      <td>1.13</td>\n",
       "      <td>378.36</td>\n",
       "      <td>1.48</td>\n",
       "      <td>1.59</td>\n",
       "      <td>10.92</td>\n",
       "      <td>34.22</td>\n",
       "      <td>...</td>\n",
       "      <td>1.90</td>\n",
       "      <td>0.23</td>\n",
       "      <td>4.99</td>\n",
       "      <td>2.19</td>\n",
       "      <td>3.54</td>\n",
       "      <td>0.06</td>\n",
       "      <td>2.33</td>\n",
       "      <td>10.21</td>\n",
       "      <td>0.93</td>\n",
       "      <td>69.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_759</th>\n",
       "      <td>0.00</td>\n",
       "      <td>31.06</td>\n",
       "      <td>8.86</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.22</td>\n",
       "      <td>475.12</td>\n",
       "      <td>1.66</td>\n",
       "      <td>0.56</td>\n",
       "      <td>10.35</td>\n",
       "      <td>25.71</td>\n",
       "      <td>...</td>\n",
       "      <td>0.06</td>\n",
       "      <td>0.01</td>\n",
       "      <td>7.41</td>\n",
       "      <td>6.63</td>\n",
       "      <td>0.90</td>\n",
       "      <td>0.47</td>\n",
       "      <td>0.44</td>\n",
       "      <td>4.86</td>\n",
       "      <td>0.00</td>\n",
       "      <td>63.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_D30</th>\n",
       "      <td>0.02</td>\n",
       "      <td>27.86</td>\n",
       "      <td>7.66</td>\n",
       "      <td>0.05</td>\n",
       "      <td>0.17</td>\n",
       "      <td>284.11</td>\n",
       "      <td>1.06</td>\n",
       "      <td>0.05</td>\n",
       "      <td>3.66</td>\n",
       "      <td>14.66</td>\n",
       "      <td>...</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.46</td>\n",
       "      <td>1.49</td>\n",
       "      <td>0.20</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.17</td>\n",
       "      <td>4.99</td>\n",
       "      <td>1.75</td>\n",
       "      <td>309.64</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>325 rows × 83 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Gene_Name  EVPL   BAG3   SKIL  HTR1B  KCNJ12     APP  P2RY14  CFTR  LRRK2  \\\n",
       "Sample                                                                      \n",
       "CGGA_1001  0.00  34.89   9.44   0.09    0.02  517.02    0.61  0.03   4.09   \n",
       "CGGA_1006  2.65  26.14  17.93   1.10    1.00  905.51    1.55  1.17   2.88   \n",
       "CGGA_1007  0.22  25.38   7.79   0.00    0.18  546.25    3.12  0.02   1.15   \n",
       "CGGA_1011  0.21  42.45  17.02   0.21    0.06  310.67    1.67  0.08  10.27   \n",
       "CGGA_1015  0.02  23.33  18.76   0.29    0.23  228.95    0.32  0.01   2.46   \n",
       "...         ...    ...    ...    ...     ...     ...     ...   ...    ...   \n",
       "CGGA_243   0.00  10.89  14.30   0.00    0.03  527.60    0.95  1.76   2.25   \n",
       "CGGA_247   1.94  20.06  17.87   0.81    1.15  593.17    0.92  0.45   2.39   \n",
       "CGGA_738   2.62  29.40  12.14   1.27    1.13  378.36    1.48  1.59  10.92   \n",
       "CGGA_759   0.00  31.06   8.86   0.09    0.22  475.12    1.66  0.56  10.35   \n",
       "CGGA_D30   0.02  27.86   7.66   0.05    0.17  284.11    1.06  0.05   3.66   \n",
       "\n",
       "Gene_Name    MYC  ...  DMRT3  HNF4A  BCL2L11  IGF2BP3  ALDH8A1  ALDH1A3  \\\n",
       "Sample            ...                                                     \n",
       "CGGA_1001  34.75  ...   0.54   0.03     8.18     1.37     1.10     7.93   \n",
       "CGGA_1006  42.67  ...   2.10   0.35     4.09    12.77     2.97     1.60   \n",
       "CGGA_1007  10.16  ...   0.00   0.00    10.54     4.69     0.32     0.06   \n",
       "CGGA_1011  28.24  ...   0.00   0.00     5.65     2.03     0.86     1.89   \n",
       "CGGA_1015  15.46  ...   0.22   0.00     3.58     3.63     0.50    36.68   \n",
       "...          ...  ...    ...    ...      ...      ...      ...      ...   \n",
       "CGGA_243   36.63  ...   0.00   0.00     3.79     0.06     1.55     0.00   \n",
       "CGGA_247   31.10  ...   0.84   0.14    12.90     0.02     2.14     0.31   \n",
       "CGGA_738   34.22  ...   1.90   0.23     4.99     2.19     3.54     0.06   \n",
       "CGGA_759   25.71  ...   0.06   0.01     7.41     6.63     0.90     0.47   \n",
       "CGGA_D30   14.66  ...   0.35   0.00     3.46     1.49     0.20     0.24   \n",
       "\n",
       "Gene_Name   KRT5  PREX2  TP53TG5  TUBB4A  \n",
       "Sample                                    \n",
       "CGGA_1001   0.03   2.71     0.35   34.38  \n",
       "CGGA_1006   2.44   7.85     1.01    4.65  \n",
       "CGGA_1007   0.23  11.78     0.19   39.15  \n",
       "CGGA_1011   0.00   6.09     0.11   14.42  \n",
       "CGGA_1015   0.00   2.04     0.00    1.32  \n",
       "...          ...    ...      ...     ...  \n",
       "CGGA_243    0.00   8.55     0.00   45.83  \n",
       "CGGA_247   25.00   2.10     0.11   52.49  \n",
       "CGGA_738    2.33  10.21     0.93   69.51  \n",
       "CGGA_759    0.44   4.86     0.00   63.06  \n",
       "CGGA_D30    0.17   4.99     1.75  309.64  \n",
       "\n",
       "[325 rows x 83 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_cgga_data_mRNAseq = cgga_data_mRNAseq[genes_in_cgga_data_mRNAseq]\n",
    "# filtered_cgga_data_mRNAseq = np.exp(filtered_cgga_data_mRNAseq_pre_log)\n",
    "filtered_cgga_data_mRNAseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "f47918d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PRS_type</th>\n",
       "      <th>Histology</th>\n",
       "      <th>Grade</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age</th>\n",
       "      <th>OS</th>\n",
       "      <th>Censor (alive=0; dead=1)</th>\n",
       "      <th>Radio_status (treated=1;un-treated=0)</th>\n",
       "      <th>Chemo_status (TMZ treated=1;un-treated=0)</th>\n",
       "      <th>IDH_mutation_status</th>\n",
       "      <th>1p19q_codeletion_status</th>\n",
       "      <th>MGMTp_methylation_status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_1001</th>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>11</td>\n",
       "      <td>3817.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1006</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>42</td>\n",
       "      <td>254.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1007</th>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Female</td>\n",
       "      <td>57</td>\n",
       "      <td>345.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1011</th>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Female</td>\n",
       "      <td>46</td>\n",
       "      <td>109.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1015</th>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>62</td>\n",
       "      <td>164.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_243</th>\n",
       "      <td>Primary</td>\n",
       "      <td>O</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Female</td>\n",
       "      <td>30</td>\n",
       "      <td>2977.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Codel</td>\n",
       "      <td>methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_247</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>60</td>\n",
       "      <td>609.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_738</th>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>43</td>\n",
       "      <td>4132.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_759</th>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>30</td>\n",
       "      <td>1263.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_D30</th>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>70</td>\n",
       "      <td>215.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>325 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          PRS_type Histology    Grade  Gender  Age      OS  \\\n",
       "CGGA_ID                                                      \n",
       "CGGA_1001  Primary       GBM   WHO IV    Male   11  3817.0   \n",
       "CGGA_1006  Primary        AA  WHO III    Male   42   254.0   \n",
       "CGGA_1007  Primary       GBM   WHO IV  Female   57   345.0   \n",
       "CGGA_1011  Primary       GBM   WHO IV  Female   46   109.0   \n",
       "CGGA_1015  Primary       GBM   WHO IV    Male   62   164.0   \n",
       "...            ...       ...      ...     ...  ...     ...   \n",
       "CGGA_243   Primary         O   WHO II  Female   30  2977.0   \n",
       "CGGA_247   Primary        AA  WHO III    Male   60   609.0   \n",
       "CGGA_738   Primary         A   WHO II    Male   43  4132.0   \n",
       "CGGA_759   Primary       GBM   WHO IV    Male   30  1263.0   \n",
       "CGGA_D30   Primary       GBM   WHO IV    Male   70   215.0   \n",
       "\n",
       "           Censor (alive=0; dead=1)  Radio_status (treated=1;un-treated=0)  \\\n",
       "CGGA_ID                                                                      \n",
       "CGGA_1001                       0.0                                    0.0   \n",
       "CGGA_1006                       1.0                                    1.0   \n",
       "CGGA_1007                       1.0                                    1.0   \n",
       "CGGA_1011                       1.0                                    1.0   \n",
       "CGGA_1015                       1.0                                    1.0   \n",
       "...                             ...                                    ...   \n",
       "CGGA_243                        0.0                                    1.0   \n",
       "CGGA_247                        1.0                                    1.0   \n",
       "CGGA_738                        0.0                                    1.0   \n",
       "CGGA_759                        1.0                                    1.0   \n",
       "CGGA_D30                        1.0                                    1.0   \n",
       "\n",
       "           Chemo_status (TMZ treated=1;un-treated=0) IDH_mutation_status  \\\n",
       "CGGA_ID                                                                    \n",
       "CGGA_1001                                        1.0            Wildtype   \n",
       "CGGA_1006                                        1.0            Wildtype   \n",
       "CGGA_1007                                        1.0            Wildtype   \n",
       "CGGA_1011                                        0.0            Wildtype   \n",
       "CGGA_1015                                        0.0            Wildtype   \n",
       "...                                              ...                 ...   \n",
       "CGGA_243                                         0.0              Mutant   \n",
       "CGGA_247                                         1.0              Mutant   \n",
       "CGGA_738                                         1.0              Mutant   \n",
       "CGGA_759                                         0.0              Mutant   \n",
       "CGGA_D30                                         0.0            Wildtype   \n",
       "\n",
       "          1p19q_codeletion_status MGMTp_methylation_status  \n",
       "CGGA_ID                                                     \n",
       "CGGA_1001               Non-codel            un-methylated  \n",
       "CGGA_1006               Non-codel            un-methylated  \n",
       "CGGA_1007               Non-codel            un-methylated  \n",
       "CGGA_1011               Non-codel            un-methylated  \n",
       "CGGA_1015               Non-codel            un-methylated  \n",
       "...                           ...                      ...  \n",
       "CGGA_243                    Codel               methylated  \n",
       "CGGA_247                Non-codel               methylated  \n",
       "CGGA_738                Non-codel            un-methylated  \n",
       "CGGA_759                Non-codel               methylated  \n",
       "CGGA_D30                Non-codel                      NaN  \n",
       "\n",
       "[325 rows x 12 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cgga_data_mRNAseq_pheno = pd.read_csv(\"CGGA.mRNAseq_325_clinical.20200506.txt\", sep='\\t')\n",
    "# cgga_data_mRNAseq_pheno\n",
    "cgga_data_mRNAseq_pheno.set_index(\"CGGA_ID\", inplace=True)\n",
    "cgga_data_mRNAseq_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "982c3fb5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>prs_type</th>\n",
       "      <th>histology</th>\n",
       "      <th>grade</th>\n",
       "      <th>gender</th>\n",
       "      <th>age_at_initial_pathologic_diagnosis</th>\n",
       "      <th>time</th>\n",
       "      <th>alive</th>\n",
       "      <th>radio_status (treated=1;un-treated=0)</th>\n",
       "      <th>chemo_status (tmz treated=1;un-treated=0)</th>\n",
       "      <th>idh_mutation_status</th>\n",
       "      <th>1p19q_codeletion_status</th>\n",
       "      <th>mgmtp_methylation_status</th>\n",
       "      <th>bcr_patient_barcode</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_1001</th>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>11</td>\n",
       "      <td>3817.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>un-methylated</td>\n",
       "      <td>CGGA_1001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1006</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>42</td>\n",
       "      <td>254.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>un-methylated</td>\n",
       "      <td>CGGA_1006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1007</th>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Female</td>\n",
       "      <td>57</td>\n",
       "      <td>345.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>un-methylated</td>\n",
       "      <td>CGGA_1007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1011</th>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Female</td>\n",
       "      <td>46</td>\n",
       "      <td>109.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>un-methylated</td>\n",
       "      <td>CGGA_1011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1015</th>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>62</td>\n",
       "      <td>164.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>un-methylated</td>\n",
       "      <td>CGGA_1015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_243</th>\n",
       "      <td>Primary</td>\n",
       "      <td>O</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Female</td>\n",
       "      <td>30</td>\n",
       "      <td>2977.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Codel</td>\n",
       "      <td>methylated</td>\n",
       "      <td>CGGA_243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_247</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>60</td>\n",
       "      <td>609.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>methylated</td>\n",
       "      <td>CGGA_247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_738</th>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>43</td>\n",
       "      <td>4132.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>un-methylated</td>\n",
       "      <td>CGGA_738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_759</th>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>30</td>\n",
       "      <td>1263.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>methylated</td>\n",
       "      <td>CGGA_759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_D30</th>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>70</td>\n",
       "      <td>215.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGGA_D30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>325 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          prs_type histology    grade  gender  \\\n",
       "CGGA_ID                                         \n",
       "CGGA_1001  Primary       GBM   WHO IV    Male   \n",
       "CGGA_1006  Primary        AA  WHO III    Male   \n",
       "CGGA_1007  Primary       GBM   WHO IV  Female   \n",
       "CGGA_1011  Primary       GBM   WHO IV  Female   \n",
       "CGGA_1015  Primary       GBM   WHO IV    Male   \n",
       "...            ...       ...      ...     ...   \n",
       "CGGA_243   Primary         O   WHO II  Female   \n",
       "CGGA_247   Primary        AA  WHO III    Male   \n",
       "CGGA_738   Primary         A   WHO II    Male   \n",
       "CGGA_759   Primary       GBM   WHO IV    Male   \n",
       "CGGA_D30   Primary       GBM   WHO IV    Male   \n",
       "\n",
       "           age_at_initial_pathologic_diagnosis    time  alive  \\\n",
       "CGGA_ID                                                         \n",
       "CGGA_1001                                   11  3817.0    0.0   \n",
       "CGGA_1006                                   42   254.0    1.0   \n",
       "CGGA_1007                                   57   345.0    1.0   \n",
       "CGGA_1011                                   46   109.0    1.0   \n",
       "CGGA_1015                                   62   164.0    1.0   \n",
       "...                                        ...     ...    ...   \n",
       "CGGA_243                                    30  2977.0    0.0   \n",
       "CGGA_247                                    60   609.0    1.0   \n",
       "CGGA_738                                    43  4132.0    0.0   \n",
       "CGGA_759                                    30  1263.0    1.0   \n",
       "CGGA_D30                                    70   215.0    1.0   \n",
       "\n",
       "           radio_status (treated=1;un-treated=0)  \\\n",
       "CGGA_ID                                            \n",
       "CGGA_1001                                    0.0   \n",
       "CGGA_1006                                    1.0   \n",
       "CGGA_1007                                    1.0   \n",
       "CGGA_1011                                    1.0   \n",
       "CGGA_1015                                    1.0   \n",
       "...                                          ...   \n",
       "CGGA_243                                     1.0   \n",
       "CGGA_247                                     1.0   \n",
       "CGGA_738                                     1.0   \n",
       "CGGA_759                                     1.0   \n",
       "CGGA_D30                                     1.0   \n",
       "\n",
       "           chemo_status (tmz treated=1;un-treated=0) idh_mutation_status  \\\n",
       "CGGA_ID                                                                    \n",
       "CGGA_1001                                        1.0            Wildtype   \n",
       "CGGA_1006                                        1.0            Wildtype   \n",
       "CGGA_1007                                        1.0            Wildtype   \n",
       "CGGA_1011                                        0.0            Wildtype   \n",
       "CGGA_1015                                        0.0            Wildtype   \n",
       "...                                              ...                 ...   \n",
       "CGGA_243                                         0.0              Mutant   \n",
       "CGGA_247                                         1.0              Mutant   \n",
       "CGGA_738                                         1.0              Mutant   \n",
       "CGGA_759                                         0.0              Mutant   \n",
       "CGGA_D30                                         0.0            Wildtype   \n",
       "\n",
       "          1p19q_codeletion_status mgmtp_methylation_status bcr_patient_barcode  \n",
       "CGGA_ID                                                                         \n",
       "CGGA_1001               Non-codel            un-methylated           CGGA_1001  \n",
       "CGGA_1006               Non-codel            un-methylated           CGGA_1006  \n",
       "CGGA_1007               Non-codel            un-methylated           CGGA_1007  \n",
       "CGGA_1011               Non-codel            un-methylated           CGGA_1011  \n",
       "CGGA_1015               Non-codel            un-methylated           CGGA_1015  \n",
       "...                           ...                      ...                 ...  \n",
       "CGGA_243                    Codel               methylated            CGGA_243  \n",
       "CGGA_247                Non-codel               methylated            CGGA_247  \n",
       "CGGA_738                Non-codel            un-methylated            CGGA_738  \n",
       "CGGA_759                Non-codel               methylated            CGGA_759  \n",
       "CGGA_D30                Non-codel                      NaN            CGGA_D30  \n",
       "\n",
       "[325 rows x 13 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cgga_data_mRNAseq_pheno.rename(columns=lambda x: x.lower(), inplace=True)\n",
    "cgga_data_mRNAseq_pheno.rename(columns={'os': 'time', 'age': 'age_at_initial_pathologic_diagnosis', \"censor (alive=0; dead=1)\": \"alive\"}, inplace=True)\n",
    "# cgga_data_mRNAseq_pheno[\"time\"] = cgga_data_pheno[\"time\"]*30\n",
    "cgga_data_mRNAseq_pheno[\"bcr_patient_barcode\"] = cgga_data_mRNAseq_pheno.index\n",
    "cgga_data_mRNAseq_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "5bf9d321",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['WHO IV', 'WHO III', 'WHO II', nan], dtype=object)"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cgga_data_mRNAseq_pheno[\"grade\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "947d7330",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>prs_type</th>\n",
       "      <th>histology</th>\n",
       "      <th>grade</th>\n",
       "      <th>gender</th>\n",
       "      <th>age_at_initial_pathologic_diagnosis</th>\n",
       "      <th>time</th>\n",
       "      <th>alive</th>\n",
       "      <th>radio_status (treated=1;un-treated=0)</th>\n",
       "      <th>chemo_status (tmz treated=1;un-treated=0)</th>\n",
       "      <th>idh_mutation_status</th>\n",
       "      <th>...</th>\n",
       "      <th>DMRT3</th>\n",
       "      <th>HNF4A</th>\n",
       "      <th>BCL2L11</th>\n",
       "      <th>IGF2BP3</th>\n",
       "      <th>ALDH8A1</th>\n",
       "      <th>ALDH1A3</th>\n",
       "      <th>KRT5</th>\n",
       "      <th>PREX2</th>\n",
       "      <th>TP53TG5</th>\n",
       "      <th>TUBB4A</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_1006</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>42</td>\n",
       "      <td>254.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>...</td>\n",
       "      <td>2.10</td>\n",
       "      <td>0.35</td>\n",
       "      <td>4.09</td>\n",
       "      <td>12.77</td>\n",
       "      <td>2.97</td>\n",
       "      <td>1.60</td>\n",
       "      <td>2.44</td>\n",
       "      <td>7.85</td>\n",
       "      <td>1.01</td>\n",
       "      <td>4.65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1022</th>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rA</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Female</td>\n",
       "      <td>62</td>\n",
       "      <td>518.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>6.44</td>\n",
       "      <td>3.10</td>\n",
       "      <td>1.46</td>\n",
       "      <td>2.15</td>\n",
       "      <td>0.11</td>\n",
       "      <td>4.07</td>\n",
       "      <td>1.18</td>\n",
       "      <td>73.53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1034</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>23</td>\n",
       "      <td>753.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>5.10</td>\n",
       "      <td>2.53</td>\n",
       "      <td>0.76</td>\n",
       "      <td>1.56</td>\n",
       "      <td>0.10</td>\n",
       "      <td>1.91</td>\n",
       "      <td>0.88</td>\n",
       "      <td>48.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1052</th>\n",
       "      <td>Primary</td>\n",
       "      <td>O</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>...</td>\n",
       "      <td>0.14</td>\n",
       "      <td>0.02</td>\n",
       "      <td>3.18</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.61</td>\n",
       "      <td>0.13</td>\n",
       "      <td>0.75</td>\n",
       "      <td>7.07</td>\n",
       "      <td>2.21</td>\n",
       "      <td>95.21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1079</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>38</td>\n",
       "      <td>471.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>...</td>\n",
       "      <td>0.03</td>\n",
       "      <td>0.00</td>\n",
       "      <td>5.91</td>\n",
       "      <td>0.39</td>\n",
       "      <td>0.62</td>\n",
       "      <td>4.31</td>\n",
       "      <td>0.35</td>\n",
       "      <td>5.76</td>\n",
       "      <td>0.58</td>\n",
       "      <td>40.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1118</th>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rAA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>31</td>\n",
       "      <td>398.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>5.12</td>\n",
       "      <td>1.28</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.24</td>\n",
       "      <td>0.88</td>\n",
       "      <td>2.41</td>\n",
       "      <td>0.15</td>\n",
       "      <td>33.21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1246</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AO</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>53</td>\n",
       "      <td>2972.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>...</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.90</td>\n",
       "      <td>0.02</td>\n",
       "      <td>1.15</td>\n",
       "      <td>0.23</td>\n",
       "      <td>0.28</td>\n",
       "      <td>5.33</td>\n",
       "      <td>1.59</td>\n",
       "      <td>88.95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_243</th>\n",
       "      <td>Primary</td>\n",
       "      <td>O</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Female</td>\n",
       "      <td>30</td>\n",
       "      <td>2977.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.79</td>\n",
       "      <td>0.06</td>\n",
       "      <td>1.55</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>8.55</td>\n",
       "      <td>0.00</td>\n",
       "      <td>45.83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_247</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>60</td>\n",
       "      <td>609.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>...</td>\n",
       "      <td>0.84</td>\n",
       "      <td>0.14</td>\n",
       "      <td>12.90</td>\n",
       "      <td>0.02</td>\n",
       "      <td>2.14</td>\n",
       "      <td>0.31</td>\n",
       "      <td>25.00</td>\n",
       "      <td>2.10</td>\n",
       "      <td>0.11</td>\n",
       "      <td>52.49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_738</th>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>43</td>\n",
       "      <td>4132.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>...</td>\n",
       "      <td>1.90</td>\n",
       "      <td>0.23</td>\n",
       "      <td>4.99</td>\n",
       "      <td>2.19</td>\n",
       "      <td>3.54</td>\n",
       "      <td>0.06</td>\n",
       "      <td>2.33</td>\n",
       "      <td>10.21</td>\n",
       "      <td>0.93</td>\n",
       "      <td>69.51</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>182 rows × 96 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            prs_type histology    grade  gender  \\\n",
       "CGGA_1006    Primary        AA  WHO III    Male   \n",
       "CGGA_1022  Recurrent        rA   WHO II  Female   \n",
       "CGGA_1034    Primary        AA  WHO III    Male   \n",
       "CGGA_1052    Primary         O   WHO II    Male   \n",
       "CGGA_1079    Primary        AA  WHO III    Male   \n",
       "...              ...       ...      ...     ...   \n",
       "CGGA_1118  Recurrent       rAA  WHO III    Male   \n",
       "CGGA_1246    Primary        AO  WHO III    Male   \n",
       "CGGA_243     Primary         O   WHO II  Female   \n",
       "CGGA_247     Primary        AA  WHO III    Male   \n",
       "CGGA_738     Primary         A   WHO II    Male   \n",
       "\n",
       "           age_at_initial_pathologic_diagnosis    time  alive  \\\n",
       "CGGA_1006                                   42   254.0    1.0   \n",
       "CGGA_1022                                   62   518.0    0.0   \n",
       "CGGA_1034                                   23   753.0    1.0   \n",
       "CGGA_1052                                   41     NaN    NaN   \n",
       "CGGA_1079                                   38   471.0    1.0   \n",
       "...                                        ...     ...    ...   \n",
       "CGGA_1118                                   31   398.0    1.0   \n",
       "CGGA_1246                                   53  2972.0    0.0   \n",
       "CGGA_243                                    30  2977.0    0.0   \n",
       "CGGA_247                                    60   609.0    1.0   \n",
       "CGGA_738                                    43  4132.0    0.0   \n",
       "\n",
       "           radio_status (treated=1;un-treated=0)  \\\n",
       "CGGA_1006                                    1.0   \n",
       "CGGA_1022                                    0.0   \n",
       "CGGA_1034                                    1.0   \n",
       "CGGA_1052                                    NaN   \n",
       "CGGA_1079                                    NaN   \n",
       "...                                          ...   \n",
       "CGGA_1118                                    0.0   \n",
       "CGGA_1246                                    1.0   \n",
       "CGGA_243                                     1.0   \n",
       "CGGA_247                                     1.0   \n",
       "CGGA_738                                     1.0   \n",
       "\n",
       "           chemo_status (tmz treated=1;un-treated=0) idh_mutation_status  ...  \\\n",
       "CGGA_1006                                        1.0            Wildtype  ...   \n",
       "CGGA_1022                                        1.0            Wildtype  ...   \n",
       "CGGA_1034                                        1.0            Wildtype  ...   \n",
       "CGGA_1052                                        NaN              Mutant  ...   \n",
       "CGGA_1079                                        NaN            Wildtype  ...   \n",
       "...                                              ...                 ...  ...   \n",
       "CGGA_1118                                        0.0              Mutant  ...   \n",
       "CGGA_1246                                        1.0              Mutant  ...   \n",
       "CGGA_243                                         0.0              Mutant  ...   \n",
       "CGGA_247                                         1.0              Mutant  ...   \n",
       "CGGA_738                                         1.0              Mutant  ...   \n",
       "\n",
       "          DMRT3 HNF4A BCL2L11  IGF2BP3  ALDH8A1  ALDH1A3   KRT5  PREX2  \\\n",
       "CGGA_1006  2.10  0.35    4.09    12.77     2.97     1.60   2.44   7.85   \n",
       "CGGA_1022  0.00  0.00    6.44     3.10     1.46     2.15   0.11   4.07   \n",
       "CGGA_1034  0.00  0.00    5.10     2.53     0.76     1.56   0.10   1.91   \n",
       "CGGA_1052  0.14  0.02    3.18     0.33     0.61     0.13   0.75   7.07   \n",
       "CGGA_1079  0.03  0.00    5.91     0.39     0.62     4.31   0.35   5.76   \n",
       "...         ...   ...     ...      ...      ...      ...    ...    ...   \n",
       "CGGA_1118  0.00  0.02    5.12     1.28     0.23     0.24   0.88   2.41   \n",
       "CGGA_1246  0.02  0.00    3.90     0.02     1.15     0.23   0.28   5.33   \n",
       "CGGA_243   0.00  0.00    3.79     0.06     1.55     0.00   0.00   8.55   \n",
       "CGGA_247   0.84  0.14   12.90     0.02     2.14     0.31  25.00   2.10   \n",
       "CGGA_738   1.90  0.23    4.99     2.19     3.54     0.06   2.33  10.21   \n",
       "\n",
       "           TP53TG5  TUBB4A  \n",
       "CGGA_1006     1.01    4.65  \n",
       "CGGA_1022     1.18   73.53  \n",
       "CGGA_1034     0.88   48.50  \n",
       "CGGA_1052     2.21   95.21  \n",
       "CGGA_1079     0.58   40.18  \n",
       "...            ...     ...  \n",
       "CGGA_1118     0.15   33.21  \n",
       "CGGA_1246     1.59   88.95  \n",
       "CGGA_243      0.00   45.83  \n",
       "CGGA_247      0.11   52.49  \n",
       "CGGA_738      0.93   69.51  \n",
       "\n",
       "[182 rows x 96 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_cgga_data_mRNAseq_with_pheno_data = pd.concat([cgga_data_mRNAseq_pheno, filtered_cgga_data_mRNAseq], axis=1)\n",
    "filtered_cgga_data_mRNAseq_with_pheno_data = filtered_cgga_data_mRNAseq_with_pheno_data[(filtered_cgga_data_mRNAseq_with_pheno_data[\"grade\"] == \"WHO III\") | (filtered_cgga_data_mRNAseq_with_pheno_data[\"grade\"] == \"WHO II\")]\n",
    "filtered_cgga_data_mRNAseq_with_pheno_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "8e2cb8a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>prs_type</th>\n",
       "      <th>histology</th>\n",
       "      <th>grade</th>\n",
       "      <th>gender</th>\n",
       "      <th>age_at_initial_pathologic_diagnosis</th>\n",
       "      <th>time</th>\n",
       "      <th>alive</th>\n",
       "      <th>radio_status (treated=1;un-treated=0)</th>\n",
       "      <th>chemo_status (tmz treated=1;un-treated=0)</th>\n",
       "      <th>idh_mutation_status</th>\n",
       "      <th>...</th>\n",
       "      <th>DMRT3median</th>\n",
       "      <th>HNF4Amedian</th>\n",
       "      <th>BCL2L11median</th>\n",
       "      <th>IGF2BP3median</th>\n",
       "      <th>ALDH8A1median</th>\n",
       "      <th>ALDH1A3median</th>\n",
       "      <th>KRT5median</th>\n",
       "      <th>PREX2median</th>\n",
       "      <th>TP53TG5median</th>\n",
       "      <th>TUBB4Amedian</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_1006</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>42</td>\n",
       "      <td>254.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1022</th>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rA</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Female</td>\n",
       "      <td>62</td>\n",
       "      <td>518.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1034</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>23</td>\n",
       "      <td>753.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1052</th>\n",
       "      <td>Primary</td>\n",
       "      <td>O</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1079</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>38</td>\n",
       "      <td>471.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1118</th>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rAA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>31</td>\n",
       "      <td>398.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1246</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AO</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>53</td>\n",
       "      <td>2972.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_243</th>\n",
       "      <td>Primary</td>\n",
       "      <td>O</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Female</td>\n",
       "      <td>30</td>\n",
       "      <td>2977.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_247</th>\n",
       "      <td>Primary</td>\n",
       "      <td>AA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>60</td>\n",
       "      <td>609.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_738</th>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>43</td>\n",
       "      <td>4132.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>182 rows × 179 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            prs_type histology    grade  gender  \\\n",
       "CGGA_1006    Primary        AA  WHO III    Male   \n",
       "CGGA_1022  Recurrent        rA   WHO II  Female   \n",
       "CGGA_1034    Primary        AA  WHO III    Male   \n",
       "CGGA_1052    Primary         O   WHO II    Male   \n",
       "CGGA_1079    Primary        AA  WHO III    Male   \n",
       "...              ...       ...      ...     ...   \n",
       "CGGA_1118  Recurrent       rAA  WHO III    Male   \n",
       "CGGA_1246    Primary        AO  WHO III    Male   \n",
       "CGGA_243     Primary         O   WHO II  Female   \n",
       "CGGA_247     Primary        AA  WHO III    Male   \n",
       "CGGA_738     Primary         A   WHO II    Male   \n",
       "\n",
       "           age_at_initial_pathologic_diagnosis    time  alive  \\\n",
       "CGGA_1006                                   42   254.0    1.0   \n",
       "CGGA_1022                                   62   518.0    0.0   \n",
       "CGGA_1034                                   23   753.0    1.0   \n",
       "CGGA_1052                                   41     NaN    NaN   \n",
       "CGGA_1079                                   38   471.0    1.0   \n",
       "...                                        ...     ...    ...   \n",
       "CGGA_1118                                   31   398.0    1.0   \n",
       "CGGA_1246                                   53  2972.0    0.0   \n",
       "CGGA_243                                    30  2977.0    0.0   \n",
       "CGGA_247                                    60   609.0    1.0   \n",
       "CGGA_738                                    43  4132.0    0.0   \n",
       "\n",
       "           radio_status (treated=1;un-treated=0)  \\\n",
       "CGGA_1006                                    1.0   \n",
       "CGGA_1022                                    0.0   \n",
       "CGGA_1034                                    1.0   \n",
       "CGGA_1052                                    NaN   \n",
       "CGGA_1079                                    NaN   \n",
       "...                                          ...   \n",
       "CGGA_1118                                    0.0   \n",
       "CGGA_1246                                    1.0   \n",
       "CGGA_243                                     1.0   \n",
       "CGGA_247                                     1.0   \n",
       "CGGA_738                                     1.0   \n",
       "\n",
       "           chemo_status (tmz treated=1;un-treated=0) idh_mutation_status  ...  \\\n",
       "CGGA_1006                                        1.0            Wildtype  ...   \n",
       "CGGA_1022                                        1.0            Wildtype  ...   \n",
       "CGGA_1034                                        1.0            Wildtype  ...   \n",
       "CGGA_1052                                        NaN              Mutant  ...   \n",
       "CGGA_1079                                        NaN            Wildtype  ...   \n",
       "...                                              ...                 ...  ...   \n",
       "CGGA_1118                                        0.0              Mutant  ...   \n",
       "CGGA_1246                                        1.0              Mutant  ...   \n",
       "CGGA_243                                         0.0              Mutant  ...   \n",
       "CGGA_247                                         1.0              Mutant  ...   \n",
       "CGGA_738                                         1.0              Mutant  ...   \n",
       "\n",
       "          DMRT3median HNF4Amedian BCL2L11median  IGF2BP3median  ALDH8A1median  \\\n",
       "CGGA_1006           1           1             0              1              1   \n",
       "CGGA_1022           0           0             1              1              1   \n",
       "CGGA_1034           0           0             0              1              1   \n",
       "CGGA_1052           1           1             0              0              0   \n",
       "CGGA_1079           0           0             1              0              0   \n",
       "...               ...         ...           ...            ...            ...   \n",
       "CGGA_1118           0           1             0              0              0   \n",
       "CGGA_1246           0           0             0              0              1   \n",
       "CGGA_243            0           0             0              0              1   \n",
       "CGGA_247            1           1             1              0              1   \n",
       "CGGA_738            1           1             0              1              1   \n",
       "\n",
       "           ALDH1A3median  KRT5median  PREX2median  TP53TG5median  TUBB4Amedian  \n",
       "CGGA_1006              1           1            1              1             0  \n",
       "CGGA_1022              1           0            0              1             0  \n",
       "CGGA_1034              1           0            0              1             0  \n",
       "CGGA_1052              0           1            1              1             1  \n",
       "CGGA_1079              1           0            1              0             0  \n",
       "...                  ...         ...          ...            ...           ...  \n",
       "CGGA_1118              0           1            0              0             0  \n",
       "CGGA_1246              0           0            1              1             1  \n",
       "CGGA_243               0           0            1              0             0  \n",
       "CGGA_247               0           1            0              0             0  \n",
       "CGGA_738               0           1            1              1             0  \n",
       "\n",
       "[182 rows x 179 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Calculate medians for each column\n",
    "filtered_cgga_data_mRNAseq_with_medians = filtered_cgga_data_mRNAseq_with_pheno_data.copy()\n",
    "medians = filtered_cgga_data_mRNAseq.median()\n",
    "cols = filtered_cgga_data_mRNAseq.columns\n",
    "# Create new columns for each column in the DataFrame\n",
    "for col in cols:\n",
    "    new_col_name = col + 'median'\n",
    "    filtered_cgga_data_mRNAseq_with_medians[new_col_name] = (filtered_cgga_data_mRNAseq[col] > medians[col]).astype(int)\n",
    "filtered_cgga_data_mRNAseq_with_medians"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "42cce6ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered_cgga_data_mRNAseq_with_medians.to_csv('cgga_data_mRNAseq.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "c849e991",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'WHO II', 'WHO III'}"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(filtered_cgga_data_mRNAseq_with_medians[\"grade\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "f73fb0e3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "291"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sum((filtered_cgga_with_pheno_data[\"grade\"] == \"II\").astype(int))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "f12750dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "334"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sum((filtered_cgga_with_pheno_data[\"grade\"] == \"III\").astype(int))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "40169018",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Anaplastic Astrocytoma',\n",
       " 'Anaplastic Oligoastrocytoma',\n",
       " 'Anaplastic Oligodendrolgioma',\n",
       " 'Astrocytoma',\n",
       " 'Oligoastrocytoma',\n",
       " 'Oligodendroglioma',\n",
       " nan}"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(filtered_cgga_with_pheno_data[\"histology\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01778603",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
