{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4a8d1386",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mDEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621\u001b[0m\u001b[33m\n",
      "\u001b[0mLooking in indexes: https://pypi.org/simple, https://alexander.ukhatov:****@pypi.pkg.jetbrains.space/glassdomeinc/p/ai/forecast-anomaly-data-science/simple\n",
      "Requirement already satisfied: openpyxl in /usr/local/lib/python3.9/site-packages (3.1.2)\n",
      "Requirement already satisfied: et-xmlfile in /usr/local/lib/python3.9/site-packages (from openpyxl) (1.1.0)\n",
      "\u001b[33mDEPRECATION: Configuring installation scheme with distutils config files is deprecated and will no longer work in the near future. If you are using a Homebrew or Linuxbrew Python, please see discussion at https://github.com/Homebrew/homebrew-core/issues/76621\u001b[0m\u001b[33m\n",
      "\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip available: \u001b[0m\u001b[31;49m22.3.1\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m23.2.1\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpython3.9 -m pip install --upgrade pip\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sys \n",
    "!{sys.executable} -m pip install openpyxl\n",
    "import openpyxl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "863fbb02",
   "metadata": {},
   "outputs": [],
   "source": [
    "cgga_data_mRNA_array = pd.read_csv(\"CGGA.mRNA_array_301_gene_level.20200506.txt\", sep='\\t')\n",
    "cgga_data_mRNA_array.set_index(\"Gene_Name\", inplace=True)\n",
    "cgga_data_mRNA_array = cgga_data_mRNA_array.transpose()\n",
    "cgga_data_mRNA_array.index.name = \"Sample\"\n",
    "# cgga_data.set_index(\"Sample\", inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "cb4cc122",
   "metadata": {},
   "outputs": [],
   "source": [
    "# cgga_data_mRNAseq.to_csv('cgga_data_mRNAseq.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4b36210e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene_Name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2BP1</th>\n",
       "      <th>A2LD1</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>A4GNT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>...</th>\n",
       "      <th>hCG_20426</th>\n",
       "      <th>hCG_2045089</th>\n",
       "      <th>tAKR</th>\n",
       "      <th>tcag7.1015</th>\n",
       "      <th>tcag7.1141</th>\n",
       "      <th>tcag7.1196</th>\n",
       "      <th>tcag7.1239</th>\n",
       "      <th>tcag7.1307</th>\n",
       "      <th>tcag7.873</th>\n",
       "      <th>tcag7.907</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_11</th>\n",
       "      <td>-0.360363</td>\n",
       "      <td>2.341360</td>\n",
       "      <td>0.034519</td>\n",
       "      <td>-0.913056</td>\n",
       "      <td>-2.248643</td>\n",
       "      <td>1.210290</td>\n",
       "      <td>1.309445</td>\n",
       "      <td>1.294572</td>\n",
       "      <td>0.750455</td>\n",
       "      <td>-0.680764</td>\n",
       "      <td>...</td>\n",
       "      <td>0.415934</td>\n",
       "      <td>2.081983</td>\n",
       "      <td>1.345131</td>\n",
       "      <td>-1.359147</td>\n",
       "      <td>0.155207</td>\n",
       "      <td>0.281617</td>\n",
       "      <td>0.856850</td>\n",
       "      <td>0.287431</td>\n",
       "      <td>-1.204858</td>\n",
       "      <td>-0.390600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_124</th>\n",
       "      <td>0.564952</td>\n",
       "      <td>1.170794</td>\n",
       "      <td>1.096407</td>\n",
       "      <td>0.510880</td>\n",
       "      <td>-1.875580</td>\n",
       "      <td>2.135452</td>\n",
       "      <td>4.251258</td>\n",
       "      <td>0.824811</td>\n",
       "      <td>0.292686</td>\n",
       "      <td>-0.342392</td>\n",
       "      <td>...</td>\n",
       "      <td>0.885566</td>\n",
       "      <td>3.163094</td>\n",
       "      <td>1.320978</td>\n",
       "      <td>-0.382007</td>\n",
       "      <td>0.011846</td>\n",
       "      <td>0.833125</td>\n",
       "      <td>1.067069</td>\n",
       "      <td>0.460703</td>\n",
       "      <td>1.164722</td>\n",
       "      <td>0.023735</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_126</th>\n",
       "      <td>0.304772</td>\n",
       "      <td>2.208151</td>\n",
       "      <td>0.302488</td>\n",
       "      <td>0.425367</td>\n",
       "      <td>-3.310851</td>\n",
       "      <td>1.633288</td>\n",
       "      <td>3.658627</td>\n",
       "      <td>1.675695</td>\n",
       "      <td>0.595336</td>\n",
       "      <td>-0.429984</td>\n",
       "      <td>...</td>\n",
       "      <td>0.740846</td>\n",
       "      <td>5.113004</td>\n",
       "      <td>2.815017</td>\n",
       "      <td>-0.925697</td>\n",
       "      <td>0.750832</td>\n",
       "      <td>-0.162234</td>\n",
       "      <td>1.781853</td>\n",
       "      <td>0.843023</td>\n",
       "      <td>-0.573511</td>\n",
       "      <td>0.468803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_168</th>\n",
       "      <td>0.174914</td>\n",
       "      <td>0.965229</td>\n",
       "      <td>0.094911</td>\n",
       "      <td>-0.194938</td>\n",
       "      <td>1.564424</td>\n",
       "      <td>-0.256463</td>\n",
       "      <td>1.841285</td>\n",
       "      <td>-0.031703</td>\n",
       "      <td>-0.033108</td>\n",
       "      <td>-0.343349</td>\n",
       "      <td>...</td>\n",
       "      <td>0.772748</td>\n",
       "      <td>0.472065</td>\n",
       "      <td>0.057275</td>\n",
       "      <td>-0.308669</td>\n",
       "      <td>0.207849</td>\n",
       "      <td>0.049558</td>\n",
       "      <td>1.959948</td>\n",
       "      <td>-1.181591</td>\n",
       "      <td>1.247256</td>\n",
       "      <td>-0.221845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_172</th>\n",
       "      <td>0.337038</td>\n",
       "      <td>-0.686878</td>\n",
       "      <td>-0.391238</td>\n",
       "      <td>0.436841</td>\n",
       "      <td>0.932857</td>\n",
       "      <td>-0.896066</td>\n",
       "      <td>-0.619468</td>\n",
       "      <td>0.082317</td>\n",
       "      <td>0.049383</td>\n",
       "      <td>-0.874413</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.311389</td>\n",
       "      <td>-0.099404</td>\n",
       "      <td>-0.781906</td>\n",
       "      <td>-0.340195</td>\n",
       "      <td>-0.530341</td>\n",
       "      <td>-0.024693</td>\n",
       "      <td>1.256403</td>\n",
       "      <td>-1.890431</td>\n",
       "      <td>1.421288</td>\n",
       "      <td>-0.535383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_719</th>\n",
       "      <td>-0.722863</td>\n",
       "      <td>-0.369550</td>\n",
       "      <td>2.221534</td>\n",
       "      <td>-0.227890</td>\n",
       "      <td>-0.709499</td>\n",
       "      <td>-0.042163</td>\n",
       "      <td>0.682384</td>\n",
       "      <td>0.353178</td>\n",
       "      <td>-0.405087</td>\n",
       "      <td>1.086748</td>\n",
       "      <td>...</td>\n",
       "      <td>0.419780</td>\n",
       "      <td>-0.314563</td>\n",
       "      <td>0.123461</td>\n",
       "      <td>0.067329</td>\n",
       "      <td>0.117164</td>\n",
       "      <td>0.231225</td>\n",
       "      <td>1.188410</td>\n",
       "      <td>-1.287032</td>\n",
       "      <td>1.056650</td>\n",
       "      <td>-0.184952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_798</th>\n",
       "      <td>0.339448</td>\n",
       "      <td>-0.190751</td>\n",
       "      <td>-0.532497</td>\n",
       "      <td>0.401836</td>\n",
       "      <td>-0.021969</td>\n",
       "      <td>0.028231</td>\n",
       "      <td>0.503611</td>\n",
       "      <td>0.695552</td>\n",
       "      <td>-0.160954</td>\n",
       "      <td>0.024414</td>\n",
       "      <td>...</td>\n",
       "      <td>0.257817</td>\n",
       "      <td>-0.448818</td>\n",
       "      <td>0.051798</td>\n",
       "      <td>0.903759</td>\n",
       "      <td>1.243486</td>\n",
       "      <td>0.499373</td>\n",
       "      <td>-0.270973</td>\n",
       "      <td>3.478258</td>\n",
       "      <td>-0.191378</td>\n",
       "      <td>0.478247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_822</th>\n",
       "      <td>0.727074</td>\n",
       "      <td>-1.250787</td>\n",
       "      <td>-0.202966</td>\n",
       "      <td>-1.145227</td>\n",
       "      <td>-0.416451</td>\n",
       "      <td>-0.027148</td>\n",
       "      <td>-0.960461</td>\n",
       "      <td>0.544798</td>\n",
       "      <td>0.284459</td>\n",
       "      <td>-0.166202</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.753053</td>\n",
       "      <td>-0.637651</td>\n",
       "      <td>0.109478</td>\n",
       "      <td>0.173328</td>\n",
       "      <td>-1.600468</td>\n",
       "      <td>0.145820</td>\n",
       "      <td>0.176987</td>\n",
       "      <td>7.053992</td>\n",
       "      <td>0.244346</td>\n",
       "      <td>-0.400094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_899</th>\n",
       "      <td>0.687279</td>\n",
       "      <td>-0.683125</td>\n",
       "      <td>0.497587</td>\n",
       "      <td>-0.236199</td>\n",
       "      <td>0.157047</td>\n",
       "      <td>-1.268045</td>\n",
       "      <td>0.127955</td>\n",
       "      <td>-0.861857</td>\n",
       "      <td>0.179601</td>\n",
       "      <td>0.212715</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.171984</td>\n",
       "      <td>-0.768285</td>\n",
       "      <td>-0.667328</td>\n",
       "      <td>-0.037473</td>\n",
       "      <td>-1.593354</td>\n",
       "      <td>-0.590683</td>\n",
       "      <td>-1.317791</td>\n",
       "      <td>-1.697378</td>\n",
       "      <td>-1.507077</td>\n",
       "      <td>-0.681704</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1068</th>\n",
       "      <td>-0.670908</td>\n",
       "      <td>2.395515</td>\n",
       "      <td>-0.448775</td>\n",
       "      <td>-1.149725</td>\n",
       "      <td>1.083235</td>\n",
       "      <td>1.313138</td>\n",
       "      <td>0.886581</td>\n",
       "      <td>0.108509</td>\n",
       "      <td>-0.153657</td>\n",
       "      <td>-0.386357</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.542129</td>\n",
       "      <td>0.193915</td>\n",
       "      <td>0.180824</td>\n",
       "      <td>0.107657</td>\n",
       "      <td>-0.731711</td>\n",
       "      <td>0.115064</td>\n",
       "      <td>-0.614588</td>\n",
       "      <td>2.008504</td>\n",
       "      <td>-0.949357</td>\n",
       "      <td>-0.046582</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>301 rows × 19416 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Gene_Name      A1BG      A1CF     A2BP1     A2LD1       A2M     A2ML1  \\\n",
       "Sample                                                                  \n",
       "CGGA_11   -0.360363  2.341360  0.034519 -0.913056 -2.248643  1.210290   \n",
       "CGGA_124   0.564952  1.170794  1.096407  0.510880 -1.875580  2.135452   \n",
       "CGGA_126   0.304772  2.208151  0.302488  0.425367 -3.310851  1.633288   \n",
       "CGGA_168   0.174914  0.965229  0.094911 -0.194938  1.564424 -0.256463   \n",
       "CGGA_172   0.337038 -0.686878 -0.391238  0.436841  0.932857 -0.896066   \n",
       "...             ...       ...       ...       ...       ...       ...   \n",
       "CGGA_719  -0.722863 -0.369550  2.221534 -0.227890 -0.709499 -0.042163   \n",
       "CGGA_798   0.339448 -0.190751 -0.532497  0.401836 -0.021969  0.028231   \n",
       "CGGA_822   0.727074 -1.250787 -0.202966 -1.145227 -0.416451 -0.027148   \n",
       "CGGA_899   0.687279 -0.683125  0.497587 -0.236199  0.157047 -1.268045   \n",
       "CGGA_1068 -0.670908  2.395515 -0.448775 -1.149725  1.083235  1.313138   \n",
       "\n",
       "Gene_Name    A4GALT     A4GNT      AAAS      AACS  ...  hCG_20426  \\\n",
       "Sample                                             ...              \n",
       "CGGA_11    1.309445  1.294572  0.750455 -0.680764  ...   0.415934   \n",
       "CGGA_124   4.251258  0.824811  0.292686 -0.342392  ...   0.885566   \n",
       "CGGA_126   3.658627  1.675695  0.595336 -0.429984  ...   0.740846   \n",
       "CGGA_168   1.841285 -0.031703 -0.033108 -0.343349  ...   0.772748   \n",
       "CGGA_172  -0.619468  0.082317  0.049383 -0.874413  ...  -0.311389   \n",
       "...             ...       ...       ...       ...  ...        ...   \n",
       "CGGA_719   0.682384  0.353178 -0.405087  1.086748  ...   0.419780   \n",
       "CGGA_798   0.503611  0.695552 -0.160954  0.024414  ...   0.257817   \n",
       "CGGA_822  -0.960461  0.544798  0.284459 -0.166202  ...  -0.753053   \n",
       "CGGA_899   0.127955 -0.861857  0.179601  0.212715  ...  -0.171984   \n",
       "CGGA_1068  0.886581  0.108509 -0.153657 -0.386357  ...  -0.542129   \n",
       "\n",
       "Gene_Name  hCG_2045089      tAKR  tcag7.1015  tcag7.1141  tcag7.1196  \\\n",
       "Sample                                                                 \n",
       "CGGA_11       2.081983  1.345131   -1.359147    0.155207    0.281617   \n",
       "CGGA_124      3.163094  1.320978   -0.382007    0.011846    0.833125   \n",
       "CGGA_126      5.113004  2.815017   -0.925697    0.750832   -0.162234   \n",
       "CGGA_168      0.472065  0.057275   -0.308669    0.207849    0.049558   \n",
       "CGGA_172     -0.099404 -0.781906   -0.340195   -0.530341   -0.024693   \n",
       "...                ...       ...         ...         ...         ...   \n",
       "CGGA_719     -0.314563  0.123461    0.067329    0.117164    0.231225   \n",
       "CGGA_798     -0.448818  0.051798    0.903759    1.243486    0.499373   \n",
       "CGGA_822     -0.637651  0.109478    0.173328   -1.600468    0.145820   \n",
       "CGGA_899     -0.768285 -0.667328   -0.037473   -1.593354   -0.590683   \n",
       "CGGA_1068     0.193915  0.180824    0.107657   -0.731711    0.115064   \n",
       "\n",
       "Gene_Name  tcag7.1239  tcag7.1307  tcag7.873  tcag7.907  \n",
       "Sample                                                   \n",
       "CGGA_11      0.856850    0.287431  -1.204858  -0.390600  \n",
       "CGGA_124     1.067069    0.460703   1.164722   0.023735  \n",
       "CGGA_126     1.781853    0.843023  -0.573511   0.468803  \n",
       "CGGA_168     1.959948   -1.181591   1.247256  -0.221845  \n",
       "CGGA_172     1.256403   -1.890431   1.421288  -0.535383  \n",
       "...               ...         ...        ...        ...  \n",
       "CGGA_719     1.188410   -1.287032   1.056650  -0.184952  \n",
       "CGGA_798    -0.270973    3.478258  -0.191378   0.478247  \n",
       "CGGA_822     0.176987    7.053992   0.244346  -0.400094  \n",
       "CGGA_899    -1.317791   -1.697378  -1.507077  -0.681704  \n",
       "CGGA_1068   -0.614588    2.008504  -0.949357  -0.046582  \n",
       "\n",
       "[301 rows x 19416 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cgga_data_mRNA_array"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "165bedbc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# [name for name in cgga_data.columns if \"GC0\" in name]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "66b1fa2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# \"FABP1\" in cgga_data.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "d3756ff9",
   "metadata": {},
   "outputs": [],
   "source": [
    "genes = list(set(list(pd.read_excel('gene_list.xlsx', sheet_name=0, header=None)[0]) +\\\n",
    "    list(pd.read_excel('gene_list.xlsx', sheet_name=1, header=None)[0])+\\\n",
    "    list(pd.read_excel('gene_list.xlsx', sheet_name=2, header=None)[0])))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5c05bbf9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "87 genes:  ['RAF1', 'KRT31', 'HTR1B', 'MYC', 'SLC12A7', 'TIE1', 'TP53TG5', 'APP', 'RLBP1', 'GPR3', 'SPRY4', 'AQP1', 'CFTR', 'SLC9A3R2', 'GNG12', 'SPSB1', 'ALDH8A1', 'CCM2L', 'ESR1', 'PPP5C', 'SHC3', 'KRT19', 'PTGDR2', 'MT-ND6', 'MT-ND3', 'KRT5', 'IGF2BP3', 'ITPR3', 'AKR1E2', 'SKIL', 'ETV5', 'CD3D', 'HLA-DQA1', 'ETV4', 'MET', 'FSHR', 'FABP1', 'PLP1', 'SYNJ2', 'CRHR2', 'TUBB4A', 'EVPL', 'DDRGK1', 'THBS4', 'ANXA2', 'DDIT4L', 'C11orf54', 'PENK', 'BCL2L11', 'PAX6', 'P2RY14', 'CORO6', 'DPY19L3', 'CHIT1', 'NPBWR2', 'MT-RNR2', 'MDFI', 'PREX2', 'PHLDB2', 'MYO5C', 'OXGR1', 'PSEN2', 'KCNJ12', 'GDF15', 'C1QTNF7', 'GRB14', 'BAG3', 'MPL', 'ALDH1A3', 'HNF4A', 'CKMT2', 'VAX1', 'GEM', 'PGAM2', 'THSD7A', 'LRRK2', 'ACTB', 'CBX3', 'DMRT3', 'BMP8A', 'PIM1', 'ABCC3', 'MYO1D', 'CD36', 'PTH1R', 'HTR1F', 'DLG4']\n"
     ]
    }
   ],
   "source": [
    "print(len(genes), \"genes: \", genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "6bc034f8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Not in cgga_data_mRNAseq: ['CCM2L', 'PTGDR2', 'MT-ND6', 'MT-ND3', 'PLP1', 'TUBB4A', 'MT-RNR2', 'PREX2', 'VAX1']\n"
     ]
    }
   ],
   "source": [
    "genes_in_cgga_data_mRNA_array = [g for g in genes if g in cgga_data_mRNA_array.columns]\n",
    "genes_not_in_cgga_data_mRNA_array = [g for g in genes if g not in cgga_data_mRNA_array.columns]\n",
    "print(f\"Not in cgga_data_mRNAseq: {genes_not_in_cgga_data_mRNA_array}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d0faaf80",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene_Name</th>\n",
       "      <th>RAF1</th>\n",
       "      <th>KRT31</th>\n",
       "      <th>HTR1B</th>\n",
       "      <th>MYC</th>\n",
       "      <th>SLC12A7</th>\n",
       "      <th>TIE1</th>\n",
       "      <th>TP53TG5</th>\n",
       "      <th>APP</th>\n",
       "      <th>RLBP1</th>\n",
       "      <th>GPR3</th>\n",
       "      <th>...</th>\n",
       "      <th>CBX3</th>\n",
       "      <th>DMRT3</th>\n",
       "      <th>BMP8A</th>\n",
       "      <th>PIM1</th>\n",
       "      <th>ABCC3</th>\n",
       "      <th>MYO1D</th>\n",
       "      <th>CD36</th>\n",
       "      <th>PTH1R</th>\n",
       "      <th>HTR1F</th>\n",
       "      <th>DLG4</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_11</th>\n",
       "      <td>0.199899</td>\n",
       "      <td>3.851997</td>\n",
       "      <td>7.358831</td>\n",
       "      <td>0.621552</td>\n",
       "      <td>3.023564</td>\n",
       "      <td>0.305205</td>\n",
       "      <td>4.684115</td>\n",
       "      <td>0.454925</td>\n",
       "      <td>0.666188</td>\n",
       "      <td>12.271709</td>\n",
       "      <td>...</td>\n",
       "      <td>0.773461</td>\n",
       "      <td>3.209948</td>\n",
       "      <td>2.659505</td>\n",
       "      <td>0.599413</td>\n",
       "      <td>0.798612</td>\n",
       "      <td>0.430209</td>\n",
       "      <td>0.726398</td>\n",
       "      <td>1.516504</td>\n",
       "      <td>8.688332</td>\n",
       "      <td>2.944458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_124</th>\n",
       "      <td>0.305667</td>\n",
       "      <td>2.109523</td>\n",
       "      <td>3.613475</td>\n",
       "      <td>0.743549</td>\n",
       "      <td>1.732212</td>\n",
       "      <td>0.265589</td>\n",
       "      <td>3.873058</td>\n",
       "      <td>0.213746</td>\n",
       "      <td>0.583500</td>\n",
       "      <td>5.213176</td>\n",
       "      <td>...</td>\n",
       "      <td>1.057804</td>\n",
       "      <td>0.691817</td>\n",
       "      <td>3.391958</td>\n",
       "      <td>1.050517</td>\n",
       "      <td>3.537371</td>\n",
       "      <td>1.389822</td>\n",
       "      <td>1.671876</td>\n",
       "      <td>0.769920</td>\n",
       "      <td>2.251112</td>\n",
       "      <td>2.958368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_126</th>\n",
       "      <td>0.254286</td>\n",
       "      <td>1.983284</td>\n",
       "      <td>6.217466</td>\n",
       "      <td>1.039545</td>\n",
       "      <td>2.919919</td>\n",
       "      <td>0.201940</td>\n",
       "      <td>4.662769</td>\n",
       "      <td>0.348627</td>\n",
       "      <td>1.953587</td>\n",
       "      <td>19.808462</td>\n",
       "      <td>...</td>\n",
       "      <td>0.924845</td>\n",
       "      <td>2.764288</td>\n",
       "      <td>4.556291</td>\n",
       "      <td>1.145075</td>\n",
       "      <td>1.957273</td>\n",
       "      <td>0.761817</td>\n",
       "      <td>1.711090</td>\n",
       "      <td>1.452649</td>\n",
       "      <td>11.559932</td>\n",
       "      <td>2.193937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_168</th>\n",
       "      <td>0.677747</td>\n",
       "      <td>0.970693</td>\n",
       "      <td>0.997041</td>\n",
       "      <td>0.848718</td>\n",
       "      <td>1.422612</td>\n",
       "      <td>0.988474</td>\n",
       "      <td>0.832228</td>\n",
       "      <td>1.858242</td>\n",
       "      <td>1.427304</td>\n",
       "      <td>1.960836</td>\n",
       "      <td>...</td>\n",
       "      <td>0.953996</td>\n",
       "      <td>28.732713</td>\n",
       "      <td>0.949244</td>\n",
       "      <td>1.316174</td>\n",
       "      <td>9.813239</td>\n",
       "      <td>1.297265</td>\n",
       "      <td>4.236367</td>\n",
       "      <td>0.774015</td>\n",
       "      <td>0.596814</td>\n",
       "      <td>1.124400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_172</th>\n",
       "      <td>0.627815</td>\n",
       "      <td>0.686414</td>\n",
       "      <td>0.503491</td>\n",
       "      <td>1.422180</td>\n",
       "      <td>0.748785</td>\n",
       "      <td>2.496159</td>\n",
       "      <td>0.534193</td>\n",
       "      <td>1.541014</td>\n",
       "      <td>1.494902</td>\n",
       "      <td>0.634406</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000907</td>\n",
       "      <td>0.808619</td>\n",
       "      <td>0.876517</td>\n",
       "      <td>1.851772</td>\n",
       "      <td>12.175131</td>\n",
       "      <td>1.523921</td>\n",
       "      <td>14.138322</td>\n",
       "      <td>0.344859</td>\n",
       "      <td>5.199614</td>\n",
       "      <td>1.117758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_719</th>\n",
       "      <td>1.078383</td>\n",
       "      <td>0.819754</td>\n",
       "      <td>0.924722</td>\n",
       "      <td>0.626762</td>\n",
       "      <td>0.946109</td>\n",
       "      <td>1.071683</td>\n",
       "      <td>1.024986</td>\n",
       "      <td>1.192165</td>\n",
       "      <td>1.163859</td>\n",
       "      <td>0.920590</td>\n",
       "      <td>...</td>\n",
       "      <td>0.679834</td>\n",
       "      <td>5.287280</td>\n",
       "      <td>0.857409</td>\n",
       "      <td>0.877970</td>\n",
       "      <td>0.673091</td>\n",
       "      <td>1.832909</td>\n",
       "      <td>0.566008</td>\n",
       "      <td>0.964346</td>\n",
       "      <td>5.707017</td>\n",
       "      <td>6.599730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_798</th>\n",
       "      <td>0.534766</td>\n",
       "      <td>1.058440</td>\n",
       "      <td>1.100992</td>\n",
       "      <td>0.990936</td>\n",
       "      <td>0.755055</td>\n",
       "      <td>0.120775</td>\n",
       "      <td>1.555818</td>\n",
       "      <td>0.757545</td>\n",
       "      <td>0.487500</td>\n",
       "      <td>0.910424</td>\n",
       "      <td>...</td>\n",
       "      <td>1.513305</td>\n",
       "      <td>1.120715</td>\n",
       "      <td>1.063246</td>\n",
       "      <td>0.900399</td>\n",
       "      <td>17.421750</td>\n",
       "      <td>2.329721</td>\n",
       "      <td>0.519811</td>\n",
       "      <td>0.753428</td>\n",
       "      <td>0.684451</td>\n",
       "      <td>0.405367</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_822</th>\n",
       "      <td>1.721614</td>\n",
       "      <td>1.199606</td>\n",
       "      <td>1.573803</td>\n",
       "      <td>1.094550</td>\n",
       "      <td>0.741663</td>\n",
       "      <td>0.427374</td>\n",
       "      <td>1.093925</td>\n",
       "      <td>1.847197</td>\n",
       "      <td>0.196751</td>\n",
       "      <td>1.025429</td>\n",
       "      <td>...</td>\n",
       "      <td>1.231947</td>\n",
       "      <td>0.955434</td>\n",
       "      <td>1.040018</td>\n",
       "      <td>0.692385</td>\n",
       "      <td>0.641303</td>\n",
       "      <td>0.223002</td>\n",
       "      <td>0.278961</td>\n",
       "      <td>0.632993</td>\n",
       "      <td>0.256635</td>\n",
       "      <td>2.176951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_899</th>\n",
       "      <td>1.310214</td>\n",
       "      <td>1.030919</td>\n",
       "      <td>1.337792</td>\n",
       "      <td>2.343515</td>\n",
       "      <td>1.267697</td>\n",
       "      <td>0.818146</td>\n",
       "      <td>5.043093</td>\n",
       "      <td>0.774339</td>\n",
       "      <td>87.150804</td>\n",
       "      <td>0.904263</td>\n",
       "      <td>...</td>\n",
       "      <td>1.145599</td>\n",
       "      <td>0.228287</td>\n",
       "      <td>2.101236</td>\n",
       "      <td>0.464124</td>\n",
       "      <td>1.713305</td>\n",
       "      <td>2.212856</td>\n",
       "      <td>2.419668</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.422351</td>\n",
       "      <td>1.380849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1068</th>\n",
       "      <td>0.987260</td>\n",
       "      <td>0.405042</td>\n",
       "      <td>1.331131</td>\n",
       "      <td>1.164522</td>\n",
       "      <td>1.182173</td>\n",
       "      <td>0.391089</td>\n",
       "      <td>0.993605</td>\n",
       "      <td>0.751958</td>\n",
       "      <td>2.493515</td>\n",
       "      <td>0.831962</td>\n",
       "      <td>...</td>\n",
       "      <td>1.143957</td>\n",
       "      <td>2.400887</td>\n",
       "      <td>0.313593</td>\n",
       "      <td>2.530906</td>\n",
       "      <td>26.591789</td>\n",
       "      <td>0.792495</td>\n",
       "      <td>2.465179</td>\n",
       "      <td>1.016951</td>\n",
       "      <td>0.239511</td>\n",
       "      <td>1.166196</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>301 rows × 78 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Gene_Name      RAF1     KRT31     HTR1B       MYC   SLC12A7      TIE1  \\\n",
       "Sample                                                                  \n",
       "CGGA_11    0.199899  3.851997  7.358831  0.621552  3.023564  0.305205   \n",
       "CGGA_124   0.305667  2.109523  3.613475  0.743549  1.732212  0.265589   \n",
       "CGGA_126   0.254286  1.983284  6.217466  1.039545  2.919919  0.201940   \n",
       "CGGA_168   0.677747  0.970693  0.997041  0.848718  1.422612  0.988474   \n",
       "CGGA_172   0.627815  0.686414  0.503491  1.422180  0.748785  2.496159   \n",
       "...             ...       ...       ...       ...       ...       ...   \n",
       "CGGA_719   1.078383  0.819754  0.924722  0.626762  0.946109  1.071683   \n",
       "CGGA_798   0.534766  1.058440  1.100992  0.990936  0.755055  0.120775   \n",
       "CGGA_822   1.721614  1.199606  1.573803  1.094550  0.741663  0.427374   \n",
       "CGGA_899   1.310214  1.030919  1.337792  2.343515  1.267697  0.818146   \n",
       "CGGA_1068  0.987260  0.405042  1.331131  1.164522  1.182173  0.391089   \n",
       "\n",
       "Gene_Name   TP53TG5       APP      RLBP1       GPR3  ...      CBX3      DMRT3  \\\n",
       "Sample                                               ...                        \n",
       "CGGA_11    4.684115  0.454925   0.666188  12.271709  ...  0.773461   3.209948   \n",
       "CGGA_124   3.873058  0.213746   0.583500   5.213176  ...  1.057804   0.691817   \n",
       "CGGA_126   4.662769  0.348627   1.953587  19.808462  ...  0.924845   2.764288   \n",
       "CGGA_168   0.832228  1.858242   1.427304   1.960836  ...  0.953996  28.732713   \n",
       "CGGA_172   0.534193  1.541014   1.494902   0.634406  ...  1.000907   0.808619   \n",
       "...             ...       ...        ...        ...  ...       ...        ...   \n",
       "CGGA_719   1.024986  1.192165   1.163859   0.920590  ...  0.679834   5.287280   \n",
       "CGGA_798   1.555818  0.757545   0.487500   0.910424  ...  1.513305   1.120715   \n",
       "CGGA_822   1.093925  1.847197   0.196751   1.025429  ...  1.231947   0.955434   \n",
       "CGGA_899   5.043093  0.774339  87.150804   0.904263  ...  1.145599   0.228287   \n",
       "CGGA_1068  0.993605  0.751958   2.493515   0.831962  ...  1.143957   2.400887   \n",
       "\n",
       "Gene_Name     BMP8A      PIM1      ABCC3     MYO1D       CD36     PTH1R  \\\n",
       "Sample                                                                    \n",
       "CGGA_11    2.659505  0.599413   0.798612  0.430209   0.726398  1.516504   \n",
       "CGGA_124   3.391958  1.050517   3.537371  1.389822   1.671876  0.769920   \n",
       "CGGA_126   4.556291  1.145075   1.957273  0.761817   1.711090  1.452649   \n",
       "CGGA_168   0.949244  1.316174   9.813239  1.297265   4.236367  0.774015   \n",
       "CGGA_172   0.876517  1.851772  12.175131  1.523921  14.138322  0.344859   \n",
       "...             ...       ...        ...       ...        ...       ...   \n",
       "CGGA_719   0.857409  0.877970   0.673091  1.832909   0.566008  0.964346   \n",
       "CGGA_798   1.063246  0.900399  17.421750  2.329721   0.519811  0.753428   \n",
       "CGGA_822   1.040018  0.692385   0.641303  0.223002   0.278961  0.632993   \n",
       "CGGA_899   2.101236  0.464124   1.713305  2.212856   2.419668  1.000000   \n",
       "CGGA_1068  0.313593  2.530906  26.591789  0.792495   2.465179  1.016951   \n",
       "\n",
       "Gene_Name      HTR1F      DLG4  \n",
       "Sample                          \n",
       "CGGA_11     8.688332  2.944458  \n",
       "CGGA_124    2.251112  2.958368  \n",
       "CGGA_126   11.559932  2.193937  \n",
       "CGGA_168    0.596814  1.124400  \n",
       "CGGA_172    5.199614  1.117758  \n",
       "...              ...       ...  \n",
       "CGGA_719    5.707017  6.599730  \n",
       "CGGA_798    0.684451  0.405367  \n",
       "CGGA_822    0.256635  2.176951  \n",
       "CGGA_899    0.422351  1.380849  \n",
       "CGGA_1068   0.239511  1.166196  \n",
       "\n",
       "[301 rows x 78 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_cgga_data_mRNA_array = cgga_data_mRNA_array[genes_in_cgga_data_mRNA_array]\n",
    "filtered_cgga_data_mRNA_array = np.exp(filtered_cgga_data_mRNA_array)\n",
    "filtered_cgga_data_mRNA_array"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f47918d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TCGA_subtypes</th>\n",
       "      <th>PRS_type</th>\n",
       "      <th>Histology</th>\n",
       "      <th>Grade</th>\n",
       "      <th>Gender</th>\n",
       "      <th>Age</th>\n",
       "      <th>OS</th>\n",
       "      <th>Censor (alive=0; dead=1)</th>\n",
       "      <th>Radio_status (treated=1;un-treated=0)</th>\n",
       "      <th>Chemo_status (TMZ treated=1;un-treated=0)</th>\n",
       "      <th>IDH_mutation_status</th>\n",
       "      <th>1p19q_Codeletion_status</th>\n",
       "      <th>MGMTp_methylation_status</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_11</th>\n",
       "      <td>Classical</td>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Female</td>\n",
       "      <td>57.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_124</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>53.0</td>\n",
       "      <td>414.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_126</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Female</td>\n",
       "      <td>50.0</td>\n",
       "      <td>1177.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_168</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>17.0</td>\n",
       "      <td>3086.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_172</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Female</td>\n",
       "      <td>57.0</td>\n",
       "      <td>462.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_719</th>\n",
       "      <td>Neural</td>\n",
       "      <td>Secondary</td>\n",
       "      <td>sGBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>51.0</td>\n",
       "      <td>101.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>un-methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_798</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Secondary</td>\n",
       "      <td>sGBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Female</td>\n",
       "      <td>41.0</td>\n",
       "      <td>136.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_822</th>\n",
       "      <td>Proneural</td>\n",
       "      <td>Secondary</td>\n",
       "      <td>sGBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>46.0</td>\n",
       "      <td>2237.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_899</th>\n",
       "      <td>Proneural</td>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rGBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>48.0</td>\n",
       "      <td>231.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>methylated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1068</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Secondary</td>\n",
       "      <td>sGBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>29.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>methylated</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>301 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          TCGA_subtypes   PRS_type Histology   Grade  Gender   Age      OS  \\\n",
       "CGGA_ID                                                                      \n",
       "CGGA_11       Classical    Primary       GBM  WHO IV  Female  57.0   155.0   \n",
       "CGGA_124    Mesenchymal    Primary       GBM  WHO IV    Male  53.0   414.0   \n",
       "CGGA_126    Mesenchymal    Primary       GBM  WHO IV  Female  50.0  1177.0   \n",
       "CGGA_168    Mesenchymal    Primary       GBM  WHO IV    Male  17.0  3086.0   \n",
       "CGGA_172    Mesenchymal    Primary       GBM  WHO IV  Female  57.0   462.0   \n",
       "...                 ...        ...       ...     ...     ...   ...     ...   \n",
       "CGGA_719         Neural  Secondary      sGBM  WHO IV    Male  51.0   101.0   \n",
       "CGGA_798    Mesenchymal  Secondary      sGBM  WHO IV  Female  41.0   136.0   \n",
       "CGGA_822      Proneural  Secondary      sGBM  WHO IV    Male  46.0  2237.0   \n",
       "CGGA_899      Proneural  Recurrent      rGBM  WHO IV    Male  48.0   231.0   \n",
       "CGGA_1068   Mesenchymal  Secondary      sGBM  WHO IV    Male  29.0    68.0   \n",
       "\n",
       "           Censor (alive=0; dead=1)  Radio_status (treated=1;un-treated=0)  \\\n",
       "CGGA_ID                                                                      \n",
       "CGGA_11                         1.0                                    1.0   \n",
       "CGGA_124                        1.0                                    1.0   \n",
       "CGGA_126                        1.0                                    1.0   \n",
       "CGGA_168                        0.0                                    1.0   \n",
       "CGGA_172                        1.0                                    1.0   \n",
       "...                             ...                                    ...   \n",
       "CGGA_719                        1.0                                    1.0   \n",
       "CGGA_798                        1.0                                    1.0   \n",
       "CGGA_822                        1.0                                    0.0   \n",
       "CGGA_899                        1.0                                    0.0   \n",
       "CGGA_1068                       1.0                                    0.0   \n",
       "\n",
       "           Chemo_status (TMZ treated=1;un-treated=0) IDH_mutation_status  \\\n",
       "CGGA_ID                                                                    \n",
       "CGGA_11                                          0.0            Wildtype   \n",
       "CGGA_124                                         0.0            Wildtype   \n",
       "CGGA_126                                         1.0            Wildtype   \n",
       "CGGA_168                                         1.0            Wildtype   \n",
       "CGGA_172                                         0.0            Wildtype   \n",
       "...                                              ...                 ...   \n",
       "CGGA_719                                         1.0            Wildtype   \n",
       "CGGA_798                                         0.0              Mutant   \n",
       "CGGA_822                                         1.0              Mutant   \n",
       "CGGA_899                                         1.0              Mutant   \n",
       "CGGA_1068                                        0.0              Mutant   \n",
       "\n",
       "          1p19q_Codeletion_status MGMTp_methylation_status  \n",
       "CGGA_ID                                                     \n",
       "CGGA_11                       NaN            un-methylated  \n",
       "CGGA_124                      NaN            un-methylated  \n",
       "CGGA_126                      NaN            un-methylated  \n",
       "CGGA_168                      NaN            un-methylated  \n",
       "CGGA_172                      NaN            un-methylated  \n",
       "...                           ...                      ...  \n",
       "CGGA_719                      NaN            un-methylated  \n",
       "CGGA_798                      NaN               methylated  \n",
       "CGGA_822                Non-codel               methylated  \n",
       "CGGA_899                Non-codel               methylated  \n",
       "CGGA_1068               Non-codel               methylated  \n",
       "\n",
       "[301 rows x 13 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cgga_data_mRNA_array_pheno = pd.read_csv(\"CGGA.mRNA_array_301_clinical.20200506.txt\", sep='\\t')\n",
    "# cgga_data_mRNAseq_pheno\n",
    "cgga_data_mRNA_array_pheno.set_index(\"CGGA_ID\", inplace=True)\n",
    "cgga_data_mRNA_array_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "982c3fb5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tcga_subtypes</th>\n",
       "      <th>prs_type</th>\n",
       "      <th>histology</th>\n",
       "      <th>grade</th>\n",
       "      <th>gender</th>\n",
       "      <th>age_at_initial_pathologic_diagnosis</th>\n",
       "      <th>time</th>\n",
       "      <th>alive</th>\n",
       "      <th>radio_status (treated=1;un-treated=0)</th>\n",
       "      <th>chemo_status (tmz treated=1;un-treated=0)</th>\n",
       "      <th>idh_mutation_status</th>\n",
       "      <th>1p19q_codeletion_status</th>\n",
       "      <th>mgmtp_methylation_status</th>\n",
       "      <th>bcr_patient_barcode</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_11</th>\n",
       "      <td>Classical</td>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Female</td>\n",
       "      <td>57.0</td>\n",
       "      <td>155.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>un-methylated</td>\n",
       "      <td>CGGA_11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_124</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>53.0</td>\n",
       "      <td>414.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>un-methylated</td>\n",
       "      <td>CGGA_124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_126</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Female</td>\n",
       "      <td>50.0</td>\n",
       "      <td>1177.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>un-methylated</td>\n",
       "      <td>CGGA_126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_168</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>17.0</td>\n",
       "      <td>3086.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>un-methylated</td>\n",
       "      <td>CGGA_168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_172</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Primary</td>\n",
       "      <td>GBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Female</td>\n",
       "      <td>57.0</td>\n",
       "      <td>462.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>un-methylated</td>\n",
       "      <td>CGGA_172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_719</th>\n",
       "      <td>Neural</td>\n",
       "      <td>Secondary</td>\n",
       "      <td>sGBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>51.0</td>\n",
       "      <td>101.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Wildtype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>un-methylated</td>\n",
       "      <td>CGGA_719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_798</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Secondary</td>\n",
       "      <td>sGBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Female</td>\n",
       "      <td>41.0</td>\n",
       "      <td>136.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>methylated</td>\n",
       "      <td>CGGA_798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_822</th>\n",
       "      <td>Proneural</td>\n",
       "      <td>Secondary</td>\n",
       "      <td>sGBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>46.0</td>\n",
       "      <td>2237.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>methylated</td>\n",
       "      <td>CGGA_822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_899</th>\n",
       "      <td>Proneural</td>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rGBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>48.0</td>\n",
       "      <td>231.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>methylated</td>\n",
       "      <td>CGGA_899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1068</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Secondary</td>\n",
       "      <td>sGBM</td>\n",
       "      <td>WHO IV</td>\n",
       "      <td>Male</td>\n",
       "      <td>29.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Mutant</td>\n",
       "      <td>Non-codel</td>\n",
       "      <td>methylated</td>\n",
       "      <td>CGGA_1068</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>301 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          tcga_subtypes   prs_type histology   grade  gender  \\\n",
       "CGGA_ID                                                        \n",
       "CGGA_11       Classical    Primary       GBM  WHO IV  Female   \n",
       "CGGA_124    Mesenchymal    Primary       GBM  WHO IV    Male   \n",
       "CGGA_126    Mesenchymal    Primary       GBM  WHO IV  Female   \n",
       "CGGA_168    Mesenchymal    Primary       GBM  WHO IV    Male   \n",
       "CGGA_172    Mesenchymal    Primary       GBM  WHO IV  Female   \n",
       "...                 ...        ...       ...     ...     ...   \n",
       "CGGA_719         Neural  Secondary      sGBM  WHO IV    Male   \n",
       "CGGA_798    Mesenchymal  Secondary      sGBM  WHO IV  Female   \n",
       "CGGA_822      Proneural  Secondary      sGBM  WHO IV    Male   \n",
       "CGGA_899      Proneural  Recurrent      rGBM  WHO IV    Male   \n",
       "CGGA_1068   Mesenchymal  Secondary      sGBM  WHO IV    Male   \n",
       "\n",
       "           age_at_initial_pathologic_diagnosis    time  alive  \\\n",
       "CGGA_ID                                                         \n",
       "CGGA_11                                   57.0   155.0    1.0   \n",
       "CGGA_124                                  53.0   414.0    1.0   \n",
       "CGGA_126                                  50.0  1177.0    1.0   \n",
       "CGGA_168                                  17.0  3086.0    0.0   \n",
       "CGGA_172                                  57.0   462.0    1.0   \n",
       "...                                        ...     ...    ...   \n",
       "CGGA_719                                  51.0   101.0    1.0   \n",
       "CGGA_798                                  41.0   136.0    1.0   \n",
       "CGGA_822                                  46.0  2237.0    1.0   \n",
       "CGGA_899                                  48.0   231.0    1.0   \n",
       "CGGA_1068                                 29.0    68.0    1.0   \n",
       "\n",
       "           radio_status (treated=1;un-treated=0)  \\\n",
       "CGGA_ID                                            \n",
       "CGGA_11                                      1.0   \n",
       "CGGA_124                                     1.0   \n",
       "CGGA_126                                     1.0   \n",
       "CGGA_168                                     1.0   \n",
       "CGGA_172                                     1.0   \n",
       "...                                          ...   \n",
       "CGGA_719                                     1.0   \n",
       "CGGA_798                                     1.0   \n",
       "CGGA_822                                     0.0   \n",
       "CGGA_899                                     0.0   \n",
       "CGGA_1068                                    0.0   \n",
       "\n",
       "           chemo_status (tmz treated=1;un-treated=0) idh_mutation_status  \\\n",
       "CGGA_ID                                                                    \n",
       "CGGA_11                                          0.0            Wildtype   \n",
       "CGGA_124                                         0.0            Wildtype   \n",
       "CGGA_126                                         1.0            Wildtype   \n",
       "CGGA_168                                         1.0            Wildtype   \n",
       "CGGA_172                                         0.0            Wildtype   \n",
       "...                                              ...                 ...   \n",
       "CGGA_719                                         1.0            Wildtype   \n",
       "CGGA_798                                         0.0              Mutant   \n",
       "CGGA_822                                         1.0              Mutant   \n",
       "CGGA_899                                         1.0              Mutant   \n",
       "CGGA_1068                                        0.0              Mutant   \n",
       "\n",
       "          1p19q_codeletion_status mgmtp_methylation_status bcr_patient_barcode  \n",
       "CGGA_ID                                                                         \n",
       "CGGA_11                       NaN            un-methylated             CGGA_11  \n",
       "CGGA_124                      NaN            un-methylated            CGGA_124  \n",
       "CGGA_126                      NaN            un-methylated            CGGA_126  \n",
       "CGGA_168                      NaN            un-methylated            CGGA_168  \n",
       "CGGA_172                      NaN            un-methylated            CGGA_172  \n",
       "...                           ...                      ...                 ...  \n",
       "CGGA_719                      NaN            un-methylated            CGGA_719  \n",
       "CGGA_798                      NaN               methylated            CGGA_798  \n",
       "CGGA_822                Non-codel               methylated            CGGA_822  \n",
       "CGGA_899                Non-codel               methylated            CGGA_899  \n",
       "CGGA_1068               Non-codel               methylated           CGGA_1068  \n",
       "\n",
       "[301 rows x 14 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cgga_data_mRNA_array_pheno.rename(columns=lambda x: x.lower(), inplace=True)\n",
    "cgga_data_mRNA_array_pheno.rename(columns={'os': 'time', 'age': 'age_at_initial_pathologic_diagnosis', \"censor (alive=0; dead=1)\": \"alive\"}, inplace=True)\n",
    "# cgga_data_mRNAseq_pheno[\"time\"] = cgga_data_pheno[\"time\"]*30\n",
    "cgga_data_mRNA_array_pheno[\"bcr_patient_barcode\"] = cgga_data_mRNA_array_pheno.index\n",
    "cgga_data_mRNA_array_pheno"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "5bf9d321",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['WHO IV', nan, 'WHO II', 'WHO III'], dtype=object)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cgga_data_mRNA_array_pheno[\"grade\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "947d7330",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tcga_subtypes</th>\n",
       "      <th>prs_type</th>\n",
       "      <th>histology</th>\n",
       "      <th>grade</th>\n",
       "      <th>gender</th>\n",
       "      <th>age_at_initial_pathologic_diagnosis</th>\n",
       "      <th>time</th>\n",
       "      <th>alive</th>\n",
       "      <th>radio_status (treated=1;un-treated=0)</th>\n",
       "      <th>chemo_status (tmz treated=1;un-treated=0)</th>\n",
       "      <th>...</th>\n",
       "      <th>CBX3</th>\n",
       "      <th>DMRT3</th>\n",
       "      <th>BMP8A</th>\n",
       "      <th>PIM1</th>\n",
       "      <th>ABCC3</th>\n",
       "      <th>MYO1D</th>\n",
       "      <th>CD36</th>\n",
       "      <th>PTH1R</th>\n",
       "      <th>HTR1F</th>\n",
       "      <th>DLG4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_253</th>\n",
       "      <td>Proneural</td>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Female</td>\n",
       "      <td>37.0</td>\n",
       "      <td>5159.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.203719</td>\n",
       "      <td>0.640748</td>\n",
       "      <td>0.880053</td>\n",
       "      <td>0.789379</td>\n",
       "      <td>0.469040</td>\n",
       "      <td>0.474720</td>\n",
       "      <td>3.607609</td>\n",
       "      <td>1.179021</td>\n",
       "      <td>0.251404</td>\n",
       "      <td>0.280883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_256</th>\n",
       "      <td>Neural</td>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>39.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.572473</td>\n",
       "      <td>0.256567</td>\n",
       "      <td>0.285044</td>\n",
       "      <td>0.656062</td>\n",
       "      <td>0.278466</td>\n",
       "      <td>0.288801</td>\n",
       "      <td>0.023790</td>\n",
       "      <td>1.409446</td>\n",
       "      <td>1.737696</td>\n",
       "      <td>0.772075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_258</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Female</td>\n",
       "      <td>61.0</td>\n",
       "      <td>663.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.935438</td>\n",
       "      <td>2.070326</td>\n",
       "      <td>1.174754</td>\n",
       "      <td>1.632784</td>\n",
       "      <td>26.416547</td>\n",
       "      <td>2.634743</td>\n",
       "      <td>8.826451</td>\n",
       "      <td>1.418649</td>\n",
       "      <td>0.427895</td>\n",
       "      <td>0.582213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_267</th>\n",
       "      <td>Neural</td>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1374.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.725631</td>\n",
       "      <td>0.365109</td>\n",
       "      <td>0.990404</td>\n",
       "      <td>0.575973</td>\n",
       "      <td>0.003473</td>\n",
       "      <td>1.946796</td>\n",
       "      <td>0.055533</td>\n",
       "      <td>1.805576</td>\n",
       "      <td>0.153907</td>\n",
       "      <td>2.232346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_271</th>\n",
       "      <td>Neural</td>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>42.0</td>\n",
       "      <td>1603.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.886252</td>\n",
       "      <td>0.624188</td>\n",
       "      <td>1.022516</td>\n",
       "      <td>0.857216</td>\n",
       "      <td>0.185536</td>\n",
       "      <td>2.099146</td>\n",
       "      <td>0.426690</td>\n",
       "      <td>0.752616</td>\n",
       "      <td>0.431432</td>\n",
       "      <td>0.323517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_406</th>\n",
       "      <td>Classical</td>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rAA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Female</td>\n",
       "      <td>24.0</td>\n",
       "      <td>90.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.979968</td>\n",
       "      <td>2.561884</td>\n",
       "      <td>0.849318</td>\n",
       "      <td>1.557538</td>\n",
       "      <td>2.112431</td>\n",
       "      <td>0.539465</td>\n",
       "      <td>1.655288</td>\n",
       "      <td>0.830498</td>\n",
       "      <td>0.308072</td>\n",
       "      <td>2.962871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_492</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rAA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>27.0</td>\n",
       "      <td>652.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.961422</td>\n",
       "      <td>0.239066</td>\n",
       "      <td>1.098953</td>\n",
       "      <td>1.895561</td>\n",
       "      <td>3.465391</td>\n",
       "      <td>2.412111</td>\n",
       "      <td>3.177271</td>\n",
       "      <td>1.371998</td>\n",
       "      <td>4.720574</td>\n",
       "      <td>1.237262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_704</th>\n",
       "      <td>Proneural</td>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>36.0</td>\n",
       "      <td>2572.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.807945</td>\n",
       "      <td>0.500090</td>\n",
       "      <td>0.295589</td>\n",
       "      <td>0.398736</td>\n",
       "      <td>0.061189</td>\n",
       "      <td>6.408868</td>\n",
       "      <td>0.210833</td>\n",
       "      <td>0.760453</td>\n",
       "      <td>0.111921</td>\n",
       "      <td>0.821444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1061</th>\n",
       "      <td>Proneural</td>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rAA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Female</td>\n",
       "      <td>43.0</td>\n",
       "      <td>532.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.623433</td>\n",
       "      <td>0.143627</td>\n",
       "      <td>0.088578</td>\n",
       "      <td>0.458193</td>\n",
       "      <td>0.014079</td>\n",
       "      <td>0.073240</td>\n",
       "      <td>0.471865</td>\n",
       "      <td>0.176590</td>\n",
       "      <td>0.135589</td>\n",
       "      <td>0.995726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_703</th>\n",
       "      <td>Neural</td>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rAO</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Female</td>\n",
       "      <td>31.0</td>\n",
       "      <td>1355.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.502243</td>\n",
       "      <td>25.505746</td>\n",
       "      <td>3.376710</td>\n",
       "      <td>1.106802</td>\n",
       "      <td>3.421708</td>\n",
       "      <td>2.721235</td>\n",
       "      <td>1.085164</td>\n",
       "      <td>0.874593</td>\n",
       "      <td>2.346036</td>\n",
       "      <td>7.007142</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>174 rows × 92 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          tcga_subtypes   prs_type histology    grade  gender  \\\n",
       "CGGA_253      Proneural    Primary         A   WHO II  Female   \n",
       "CGGA_256         Neural    Primary         A   WHO II    Male   \n",
       "CGGA_258    Mesenchymal    Primary         A   WHO II  Female   \n",
       "CGGA_267         Neural    Primary         A   WHO II    Male   \n",
       "CGGA_271         Neural    Primary         A   WHO II    Male   \n",
       "...                 ...        ...       ...      ...     ...   \n",
       "CGGA_406      Classical  Recurrent       rAA  WHO III  Female   \n",
       "CGGA_492    Mesenchymal  Recurrent       rAA  WHO III    Male   \n",
       "CGGA_704      Proneural    Primary         A   WHO II    Male   \n",
       "CGGA_1061     Proneural  Recurrent       rAA  WHO III  Female   \n",
       "CGGA_703         Neural  Recurrent       rAO  WHO III  Female   \n",
       "\n",
       "           age_at_initial_pathologic_diagnosis    time  alive  \\\n",
       "CGGA_253                                  37.0  5159.0    0.0   \n",
       "CGGA_256                                  39.0   158.0    0.0   \n",
       "CGGA_258                                  61.0   663.0    1.0   \n",
       "CGGA_267                                  36.0  1374.0    1.0   \n",
       "CGGA_271                                  42.0  1603.0    0.0   \n",
       "...                                        ...     ...    ...   \n",
       "CGGA_406                                  24.0    90.0    1.0   \n",
       "CGGA_492                                  27.0   652.0    0.0   \n",
       "CGGA_704                                  36.0  2572.0    1.0   \n",
       "CGGA_1061                                 43.0   532.0    1.0   \n",
       "CGGA_703                                  31.0  1355.0    1.0   \n",
       "\n",
       "           radio_status (treated=1;un-treated=0)  \\\n",
       "CGGA_253                                     1.0   \n",
       "CGGA_256                                     1.0   \n",
       "CGGA_258                                     1.0   \n",
       "CGGA_267                                     0.0   \n",
       "CGGA_271                                     1.0   \n",
       "...                                          ...   \n",
       "CGGA_406                                     0.0   \n",
       "CGGA_492                                     1.0   \n",
       "CGGA_704                                     1.0   \n",
       "CGGA_1061                                    0.0   \n",
       "CGGA_703                                     0.0   \n",
       "\n",
       "           chemo_status (tmz treated=1;un-treated=0)  ...      CBX3  \\\n",
       "CGGA_253                                         0.0  ...  1.203719   \n",
       "CGGA_256                                         1.0  ...  0.572473   \n",
       "CGGA_258                                         1.0  ...  0.935438   \n",
       "CGGA_267                                         0.0  ...  0.725631   \n",
       "CGGA_271                                         1.0  ...  0.886252   \n",
       "...                                              ...  ...       ...   \n",
       "CGGA_406                                         0.0  ...  2.979968   \n",
       "CGGA_492                                         1.0  ...  1.961422   \n",
       "CGGA_704                                         0.0  ...  0.807945   \n",
       "CGGA_1061                                        1.0  ...  1.623433   \n",
       "CGGA_703                                         1.0  ...  0.502243   \n",
       "\n",
       "               DMRT3     BMP8A      PIM1      ABCC3     MYO1D      CD36  \\\n",
       "CGGA_253    0.640748  0.880053  0.789379   0.469040  0.474720  3.607609   \n",
       "CGGA_256    0.256567  0.285044  0.656062   0.278466  0.288801  0.023790   \n",
       "CGGA_258    2.070326  1.174754  1.632784  26.416547  2.634743  8.826451   \n",
       "CGGA_267    0.365109  0.990404  0.575973   0.003473  1.946796  0.055533   \n",
       "CGGA_271    0.624188  1.022516  0.857216   0.185536  2.099146  0.426690   \n",
       "...              ...       ...       ...        ...       ...       ...   \n",
       "CGGA_406    2.561884  0.849318  1.557538   2.112431  0.539465  1.655288   \n",
       "CGGA_492    0.239066  1.098953  1.895561   3.465391  2.412111  3.177271   \n",
       "CGGA_704    0.500090  0.295589  0.398736   0.061189  6.408868  0.210833   \n",
       "CGGA_1061   0.143627  0.088578  0.458193   0.014079  0.073240  0.471865   \n",
       "CGGA_703   25.505746  3.376710  1.106802   3.421708  2.721235  1.085164   \n",
       "\n",
       "              PTH1R     HTR1F      DLG4  \n",
       "CGGA_253   1.179021  0.251404  0.280883  \n",
       "CGGA_256   1.409446  1.737696  0.772075  \n",
       "CGGA_258   1.418649  0.427895  0.582213  \n",
       "CGGA_267   1.805576  0.153907  2.232346  \n",
       "CGGA_271   0.752616  0.431432  0.323517  \n",
       "...             ...       ...       ...  \n",
       "CGGA_406   0.830498  0.308072  2.962871  \n",
       "CGGA_492   1.371998  4.720574  1.237262  \n",
       "CGGA_704   0.760453  0.111921  0.821444  \n",
       "CGGA_1061  0.176590  0.135589  0.995726  \n",
       "CGGA_703   0.874593  2.346036  7.007142  \n",
       "\n",
       "[174 rows x 92 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_cgga_data_mRNA_array_with_pheno_data = pd.concat([cgga_data_mRNA_array_pheno, filtered_cgga_data_mRNA_array], axis=1)\n",
    "filtered_cgga_data_mRNA_array_with_pheno_data = filtered_cgga_data_mRNA_array_with_pheno_data[(filtered_cgga_data_mRNA_array_with_pheno_data[\"grade\"] == \"WHO III\") | (filtered_cgga_data_mRNA_array_with_pheno_data[\"grade\"] == \"WHO II\")]\n",
    "filtered_cgga_data_mRNA_array_with_pheno_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "8e2cb8a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tcga_subtypes</th>\n",
       "      <th>prs_type</th>\n",
       "      <th>histology</th>\n",
       "      <th>grade</th>\n",
       "      <th>gender</th>\n",
       "      <th>age_at_initial_pathologic_diagnosis</th>\n",
       "      <th>time</th>\n",
       "      <th>alive</th>\n",
       "      <th>radio_status (treated=1;un-treated=0)</th>\n",
       "      <th>chemo_status (tmz treated=1;un-treated=0)</th>\n",
       "      <th>...</th>\n",
       "      <th>CBX3median</th>\n",
       "      <th>DMRT3median</th>\n",
       "      <th>BMP8Amedian</th>\n",
       "      <th>PIM1median</th>\n",
       "      <th>ABCC3median</th>\n",
       "      <th>MYO1Dmedian</th>\n",
       "      <th>CD36median</th>\n",
       "      <th>PTH1Rmedian</th>\n",
       "      <th>HTR1Fmedian</th>\n",
       "      <th>DLG4median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CGGA_253</th>\n",
       "      <td>Proneural</td>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Female</td>\n",
       "      <td>37.0</td>\n",
       "      <td>5159.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_256</th>\n",
       "      <td>Neural</td>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>39.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_258</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Female</td>\n",
       "      <td>61.0</td>\n",
       "      <td>663.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_267</th>\n",
       "      <td>Neural</td>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>36.0</td>\n",
       "      <td>1374.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_271</th>\n",
       "      <td>Neural</td>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>42.0</td>\n",
       "      <td>1603.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_406</th>\n",
       "      <td>Classical</td>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rAA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Female</td>\n",
       "      <td>24.0</td>\n",
       "      <td>90.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_492</th>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rAA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Male</td>\n",
       "      <td>27.0</td>\n",
       "      <td>652.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_704</th>\n",
       "      <td>Proneural</td>\n",
       "      <td>Primary</td>\n",
       "      <td>A</td>\n",
       "      <td>WHO II</td>\n",
       "      <td>Male</td>\n",
       "      <td>36.0</td>\n",
       "      <td>2572.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_1061</th>\n",
       "      <td>Proneural</td>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rAA</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Female</td>\n",
       "      <td>43.0</td>\n",
       "      <td>532.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGGA_703</th>\n",
       "      <td>Neural</td>\n",
       "      <td>Recurrent</td>\n",
       "      <td>rAO</td>\n",
       "      <td>WHO III</td>\n",
       "      <td>Female</td>\n",
       "      <td>31.0</td>\n",
       "      <td>1355.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>174 rows × 170 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          tcga_subtypes   prs_type histology    grade  gender  \\\n",
       "CGGA_253      Proneural    Primary         A   WHO II  Female   \n",
       "CGGA_256         Neural    Primary         A   WHO II    Male   \n",
       "CGGA_258    Mesenchymal    Primary         A   WHO II  Female   \n",
       "CGGA_267         Neural    Primary         A   WHO II    Male   \n",
       "CGGA_271         Neural    Primary         A   WHO II    Male   \n",
       "...                 ...        ...       ...      ...     ...   \n",
       "CGGA_406      Classical  Recurrent       rAA  WHO III  Female   \n",
       "CGGA_492    Mesenchymal  Recurrent       rAA  WHO III    Male   \n",
       "CGGA_704      Proneural    Primary         A   WHO II    Male   \n",
       "CGGA_1061     Proneural  Recurrent       rAA  WHO III  Female   \n",
       "CGGA_703         Neural  Recurrent       rAO  WHO III  Female   \n",
       "\n",
       "           age_at_initial_pathologic_diagnosis    time  alive  \\\n",
       "CGGA_253                                  37.0  5159.0    0.0   \n",
       "CGGA_256                                  39.0   158.0    0.0   \n",
       "CGGA_258                                  61.0   663.0    1.0   \n",
       "CGGA_267                                  36.0  1374.0    1.0   \n",
       "CGGA_271                                  42.0  1603.0    0.0   \n",
       "...                                        ...     ...    ...   \n",
       "CGGA_406                                  24.0    90.0    1.0   \n",
       "CGGA_492                                  27.0   652.0    0.0   \n",
       "CGGA_704                                  36.0  2572.0    1.0   \n",
       "CGGA_1061                                 43.0   532.0    1.0   \n",
       "CGGA_703                                  31.0  1355.0    1.0   \n",
       "\n",
       "           radio_status (treated=1;un-treated=0)  \\\n",
       "CGGA_253                                     1.0   \n",
       "CGGA_256                                     1.0   \n",
       "CGGA_258                                     1.0   \n",
       "CGGA_267                                     0.0   \n",
       "CGGA_271                                     1.0   \n",
       "...                                          ...   \n",
       "CGGA_406                                     0.0   \n",
       "CGGA_492                                     1.0   \n",
       "CGGA_704                                     1.0   \n",
       "CGGA_1061                                    0.0   \n",
       "CGGA_703                                     0.0   \n",
       "\n",
       "           chemo_status (tmz treated=1;un-treated=0)  ... CBX3median  \\\n",
       "CGGA_253                                         0.0  ...          1   \n",
       "CGGA_256                                         1.0  ...          0   \n",
       "CGGA_258                                         1.0  ...          0   \n",
       "CGGA_267                                         0.0  ...          0   \n",
       "CGGA_271                                         1.0  ...          0   \n",
       "...                                              ...  ...        ...   \n",
       "CGGA_406                                         0.0  ...          1   \n",
       "CGGA_492                                         1.0  ...          1   \n",
       "CGGA_704                                         0.0  ...          0   \n",
       "CGGA_1061                                        1.0  ...          1   \n",
       "CGGA_703                                         1.0  ...          0   \n",
       "\n",
       "          DMRT3median BMP8Amedian PIM1median  ABCC3median  MYO1Dmedian  \\\n",
       "CGGA_253            0           0          0            0            0   \n",
       "CGGA_256            0           0          0            0            0   \n",
       "CGGA_258            1           1          1            1            1   \n",
       "CGGA_267            0           0          0            0            1   \n",
       "CGGA_271            0           1          0            0            1   \n",
       "...               ...         ...        ...          ...          ...   \n",
       "CGGA_406            1           0          1            1            0   \n",
       "CGGA_492            0           1          1            1            1   \n",
       "CGGA_704            0           0          0            0            1   \n",
       "CGGA_1061           0           0          0            0            0   \n",
       "CGGA_703            1           1          1            1            1   \n",
       "\n",
       "           CD36median  PTH1Rmedian  HTR1Fmedian  DLG4median  \n",
       "CGGA_253            1            1            0           0  \n",
       "CGGA_256            0            1            1           0  \n",
       "CGGA_258            1            1            0           0  \n",
       "CGGA_267            0            1            0           1  \n",
       "CGGA_271            0            0            0           0  \n",
       "...               ...          ...          ...         ...  \n",
       "CGGA_406            1            0            0           1  \n",
       "CGGA_492            1            1            1           1  \n",
       "CGGA_704            0            0            0           0  \n",
       "CGGA_1061           0            0            0           1  \n",
       "CGGA_703            1            0            1           1  \n",
       "\n",
       "[174 rows x 170 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Calculate medians for each column\n",
    "filtered_cgga_data_mRNA_array_with_medians = filtered_cgga_data_mRNA_array_with_pheno_data.copy()\n",
    "medians = filtered_cgga_data_mRNA_array.median()\n",
    "cols = filtered_cgga_data_mRNA_array.columns\n",
    "# Create new columns for each column in the DataFrame\n",
    "for col in cols:\n",
    "    new_col_name = col + 'median'\n",
    "    filtered_cgga_data_mRNA_array_with_medians[new_col_name] = (filtered_cgga_data_mRNA_array[col] > medians[col]).astype(int)\n",
    "filtered_cgga_data_mRNA_array_with_medians"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "42cce6ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered_cgga_data_mRNA_array_with_medians.to_csv('cgga_data_mRNA_array.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "c849e991",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'WHO II', 'WHO III'}"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(filtered_cgga_data_mRNAseq_with_medians[\"grade\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "f73fb0e3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "291"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sum((filtered_cgga_with_pheno_data[\"grade\"] == \"II\").astype(int))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "f12750dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "334"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sum((filtered_cgga_with_pheno_data[\"grade\"] == \"III\").astype(int))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "40169018",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Anaplastic Astrocytoma',\n",
       " 'Anaplastic Oligoastrocytoma',\n",
       " 'Anaplastic Oligodendrolgioma',\n",
       " 'Astrocytoma',\n",
       " 'Oligoastrocytoma',\n",
       " 'Oligodendroglioma',\n",
       " nan}"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(filtered_cgga_with_pheno_data[\"histology\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01778603",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
